{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "!wget github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
        "!tar -zxvf bedtools-2.30.0.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "T-xbKpaxRizk",
        "outputId": "d80cae42-fa8f-45a7-db2f-bd7490afa37b"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-12 21:07:30--  http://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.114.4\n",
            "Connecting to github.com (github.com)|140.82.114.4|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz [following]\n",
            "--2024-06-12 21:07:30--  https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Connecting to github.com (github.com)|140.82.114.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240612%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240612T210730Z&X-Amz-Expires=300&X-Amz-Signature=56182c1eb1acf4c5cca9ae486df682627a2f583c5553f03724e0dcfef4ce4457&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2024-06-12 21:07:30--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240612%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240612T210730Z&X-Amz-Expires=300&X-Amz-Signature=56182c1eb1acf4c5cca9ae486df682627a2f583c5553f03724e0dcfef4ce4457&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 20648577 (20M) [application/octet-stream]\n",
            "Saving to: ‘bedtools-2.30.0.tar.gz’\n",
            "\n",
            "bedtools-2.30.0.tar 100%[===================>]  19.69M  95.3MB/s    in 0.2s    \n",
            "\n",
            "2024-06-12 21:07:30 (95.3 MB/s) - ‘bedtools-2.30.0.tar.gz’ saved [20648577/20648577]\n",
            "\n",
            "bedtools2/\n",
            "bedtools2/obj/\n",
            "bedtools2/tutorial/\n",
            "bedtools2/LICENSE\n",
            "bedtools2/test/\n",
            "bedtools2/bin/\n",
            "bedtools2/Makefile\n",
            "bedtools2/docs/\n",
            "bedtools2/README.md\n",
            "bedtools2/genomes/\n",
            "bedtools2/.gitignore\n",
            "bedtools2/scripts/\n",
            "bedtools2/.github/\n",
            "bedtools2/data/\n",
            "bedtools2/src/\n",
            "bedtools2/src/spacingFile/\n",
            "bedtools2/src/sortBed/\n",
            "bedtools2/src/split/\n",
            "bedtools2/src/bedToBam/\n",
            "bedtools2/src/complementFile/\n",
            "bedtools2/src/expand/\n",
            "bedtools2/src/fisher/\n",
            "bedtools2/src/linksBed/\n",
            "bedtools2/src/pairToPair/\n",
            "bedtools2/src/bedtools.cpp\n",
            "bedtools2/src/mapFile/\n",
            "bedtools2/src/slopBed/\n",
            "bedtools2/src/annotateBed/\n",
            "bedtools2/src/randomBed/\n",
            "bedtools2/src/getOverlap/\n",
            "bedtools2/src/reldist/\n",
            "bedtools2/src/nucBed/\n",
            "bedtools2/src/bedToIgv/\n",
            "bedtools2/src/regressTest/\n",
            "bedtools2/src/maskFastaFromBed/\n",
            "bedtools2/src/genomeCoverageBed/\n",
            "bedtools2/src/utils/\n",
            "bedtools2/src/windowBed/\n",
            "bedtools2/src/jaccard/\n",
            "bedtools2/src/groupBy/\n",
            "bedtools2/src/bamToFastq/\n",
            "bedtools2/src/htsutil.cpp\n",
            "bedtools2/src/fastaFromBed/\n",
            "bedtools2/src/nekSandbox1/\n",
            "bedtools2/src/unionBedGraphs/\n",
            "bedtools2/src/bed12ToBed6/\n",
            "bedtools2/src/subtractFile/\n",
            "bedtools2/src/bedpeToBam/\n",
            "bedtools2/src/windowMaker/\n",
            "bedtools2/src/coverageFile/\n",
            "bedtools2/src/summaryFile/\n",
            "bedtools2/src/multiIntersectBed/\n",
            "bedtools2/src/intersectFile/\n",
            "bedtools2/src/closestFile/\n",
            "bedtools2/src/shuffleBed/\n",
            "bedtools2/src/flankBed/\n",
            "bedtools2/src/tagBam/\n",
            "bedtools2/src/multiBamCov/\n",
            "bedtools2/src/sampleFile/\n",
            "bedtools2/src/clusterBed/\n",
            "bedtools2/src/shiftBed/\n",
            "bedtools2/src/mergeFile/\n",
            "bedtools2/src/bamToBed/\n",
            "bedtools2/src/pairToBed/\n",
            "bedtools2/src/pairToBed/pairToBed.h\n",
            "bedtools2/src/pairToBed/pairToBedMain.cpp\n",
            "bedtools2/src/pairToBed/Makefile.frag\n",
            "bedtools2/src/pairToBed/pairToBed.cpp\n",
            "bedtools2/src/bamToBed/bamToBed.cpp\n",
            "bedtools2/src/bamToBed/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeFile.cpp\n",
            "bedtools2/src/mergeFile/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeHelp.cpp\n",
            "bedtools2/src/mergeFile/mergeFile.h\n",
            "bedtools2/src/shiftBed/shiftBed.cpp\n",
            "bedtools2/src/shiftBed/shiftBedMain.cpp\n",
            "bedtools2/src/shiftBed/Makefile.frag\n",
            "bedtools2/src/shiftBed/shiftBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.cpp\n",
            "bedtools2/src/clusterBed/clusterMain.cpp\n",
            "bedtools2/src/clusterBed/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleFile.cpp\n",
            "bedtools2/src/sampleFile/sampleFile.h\n",
            "bedtools2/src/sampleFile/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleHelp.cpp\n",
            "bedtools2/src/multiBamCov/multiBamCov.cpp\n",
            "bedtools2/src/multiBamCov/Makefile.frag\n",
            "bedtools2/src/multiBamCov/multiBamCov.h\n",
            "bedtools2/src/multiBamCov/multiBamCovMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.cpp\n",
            "bedtools2/src/tagBam/tagBamMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.h\n",
            "bedtools2/src/tagBam/Makefile.frag\n",
            "bedtools2/src/flankBed/flankBed.cpp\n",
            "bedtools2/src/flankBed/flankBedMain.cpp\n",
            "bedtools2/src/flankBed/test.fa.fai\n",
            "bedtools2/src/flankBed/test.gff\n",
            "bedtools2/src/flankBed/Makefile.frag\n",
            "bedtools2/src/flankBed/test.fa\n",
            "bedtools2/src/flankBed/flankBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.cpp\n",
            "bedtools2/src/shuffleBed/Makefile.frag\n",
            "bedtools2/src/shuffleBed/shuffleBedMain.cpp\n",
            "bedtools2/src/closestFile/closestHelp.cpp\n",
            "bedtools2/src/closestFile/closestFile.h\n",
            "bedtools2/src/closestFile/closestFile.cpp\n",
            "bedtools2/src/closestFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectHelp.cpp\n",
            "bedtools2/src/intersectFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectFile.cpp\n",
            "bedtools2/src/intersectFile/intersectFile.h\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.cpp\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.h\n",
            "bedtools2/src/multiIntersectBed/Makefile.frag\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "bedtools2/src/summaryFile/summaryHelp.cpp\n",
            "bedtools2/src/summaryFile/Makefile.frag\n",
            "bedtools2/src/summaryFile/summaryFile.h\n",
            "bedtools2/src/summaryFile/summaryFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.h\n",
            "bedtools2/src/coverageFile/Makefile.frag\n",
            "bedtools2/src/coverageFile/coverageHelp.cpp\n",
            "bedtools2/src/windowMaker/windowMakerMain.cpp\n",
            "bedtools2/src/windowMaker/windowMaker.h\n",
            "bedtools2/src/windowMaker/windowMaker.cpp\n",
            "bedtools2/src/windowMaker/Makefile.frag\n",
            "bedtools2/src/bedpeToBam/notes.txt\n",
            "bedtools2/src/bedpeToBam/bedpeToBam.cpp\n",
            "bedtools2/src/bedpeToBam/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractHelp.cpp\n",
            "bedtools2/src/subtractFile/subtractFile.h\n",
            "bedtools2/src/subtractFile/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractFile.cpp\n",
            "bedtools2/src/bed12ToBed6/bed12ToBed6.cpp\n",
            "bedtools2/src/bed12ToBed6/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.cpp\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "bedtools2/src/unionBedGraphs/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.h\n",
            "bedtools2/src/nekSandbox1/FormatGuess.h\n",
            "bedtools2/src/nekSandbox1/Makefile\n",
            "bedtools2/src/nekSandbox1/nekSandboxMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBedMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.h\n",
            "bedtools2/src/fastaFromBed/Makefile.frag\n",
            "bedtools2/src/bamToFastq/bamToFastq.h\n",
            "bedtools2/src/bamToFastq/bamToFastq.cpp\n",
            "bedtools2/src/bamToFastq/bamToFastqMain.cpp\n",
            "bedtools2/src/bamToFastq/Makefile.frag\n",
            "bedtools2/src/groupBy/groupBy.cpp\n",
            "bedtools2/src/groupBy/groupBy.h\n",
            "bedtools2/src/groupBy/Makefile.frag\n",
            "bedtools2/src/groupBy/groupByHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.cpp\n",
            "bedtools2/src/jaccard/Makefile.frag\n",
            "bedtools2/src/jaccard/jaccardHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.h\n",
            "bedtools2/src/windowBed/windowBed.cpp\n",
            "bedtools2/src/windowBed/windowBed.h\n",
            "bedtools2/src/windowBed/windowMain.cpp\n",
            "bedtools2/src/windowBed/Makefile.frag\n",
            "bedtools2/src/utils/tabFile/\n",
            "bedtools2/src/utils/htslib/\n",
            "bedtools2/src/utils/NewChromsweep/\n",
            "bedtools2/src/utils/RecordOutputMgr/\n",
            "bedtools2/src/utils/bedFile/\n",
            "bedtools2/src/utils/Contexts/\n",
            "bedtools2/src/utils/BamTools/\n",
            "bedtools2/src/utils/BlockedIntervals/\n",
            "bedtools2/src/utils/general/\n",
            "bedtools2/src/utils/driver/\n",
            "bedtools2/src/utils/Point/\n",
            "bedtools2/src/utils/bedGraphFile/\n",
            "bedtools2/src/utils/sequenceUtilities/\n",
            "bedtools2/src/utils/Fasta/\n",
            "bedtools2/src/utils/gzstream/\n",
            "bedtools2/src/utils/fileType/\n",
            "bedtools2/src/utils/KeyListOps/\n",
            "bedtools2/src/utils/BamTools-Ancillary/\n",
            "bedtools2/src/utils/GenomeFile/\n",
            "bedtools2/src/utils/version/\n",
            "bedtools2/src/utils/ToolBase/\n",
            "bedtools2/src/utils/lineFileUtilities/\n",
            "bedtools2/src/utils/bedFilePE/\n",
            "bedtools2/src/utils/chromsweep/\n",
            "bedtools2/src/utils/FileRecordTools/\n",
            "bedtools2/src/utils/BinTree/\n",
            "bedtools2/src/utils/VectorOps/\n",
            "bedtools2/src/utils/stringUtilities/\n",
            "bedtools2/src/utils/stringUtilities/stringUtilities.h\n",
            "bedtools2/src/utils/VectorOps/VectorOps.cpp\n",
            "bedtools2/src/utils/VectorOps/Makefile.frag\n",
            "bedtools2/src/utils/VectorOps/VectorOps.h\n",
            "bedtools2/src/utils/BinTree/BinTree.cpp\n",
            "bedtools2/src/utils/BinTree/BinTree.h\n",
            "bedtools2/src/utils/BinTree/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
            "bedtools2/src/utils/chromsweep/chromsweep.cpp\n",
            "bedtools2/src/utils/chromsweep/Makefile.frag\n",
            "bedtools2/src/utils/chromsweep/chromsweep.h\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.cpp\n",
            "bedtools2/src/utils/bedFilePE/Makefile.frag\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.h\n",
            "bedtools2/src/utils/lineFileUtilities/lineFileUtilities.h\n",
            "bedtools2/src/utils/ToolBase/ToolBase.cpp\n",
            "bedtools2/src/utils/ToolBase/Makefile.frag\n",
            "bedtools2/src/utils/ToolBase/ToolBase.h\n",
            "bedtools2/src/utils/version/version.h\n",
            "bedtools2/src/utils/version/version.cpp\n",
            "bedtools2/src/utils/version/version_release.txt\n",
            "bedtools2/src/utils/version/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.h\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "bedtools2/src/utils/BamTools-Ancillary/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.h\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.cpp\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "bedtools2/src/utils/KeyListOps/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.h\n",
            "bedtools2/src/utils/fileType/fileType.cpp\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.h\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "bedtools2/src/utils/fileType/Makefile.frag\n",
            "bedtools2/src/utils/fileType/fileType.h\n",
            "bedtools2/src/utils/gzstream/gzstream.C\n",
            "bedtools2/src/utils/gzstream/README\n",
            "bedtools2/src/utils/gzstream/COPYING.LIB\n",
            "bedtools2/src/utils/gzstream/Makefile.frag\n",
            "bedtools2/src/utils/gzstream/version.txt\n",
            "bedtools2/src/utils/gzstream/gzstream.h\n",
            "bedtools2/src/utils/Fasta/Fasta.h\n",
            "bedtools2/src/utils/Fasta/split.cpp\n",
            "bedtools2/src/utils/Fasta/Fasta.cpp\n",
            "bedtools2/src/utils/Fasta/LargeFileSupport.h\n",
            "bedtools2/src/utils/Fasta/Makefile.frag\n",
            "bedtools2/src/utils/Fasta/split.h\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.h\n",
            "bedtools2/src/utils/sequenceUtilities/Makefile.frag\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.h\n",
            "bedtools2/src/utils/bedGraphFile/Makefile.frag\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "bedtools2/src/utils/Point/Point.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.cpp\n",
            "bedtools2/src/utils/driver/Makefile.frag\n",
            "bedtools2/src/utils/general/QuickString.cpp\n",
            "bedtools2/src/utils/general/ParseTools.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.h\n",
            "bedtools2/src/utils/general/FreeList.h\n",
            "bedtools2/src/utils/general/CompressionTools.cpp\n",
            "bedtools2/src/utils/general/Random.cpp\n",
            "bedtools2/src/utils/general/BedtoolsTypes.h\n",
            "bedtools2/src/utils/general/ErrorMsg.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.h\n",
            "bedtools2/src/utils/general/Tokenizer.h\n",
            "bedtools2/src/utils/general/ParseTools.h\n",
            "bedtools2/src/utils/general/QuickString.h\n",
            "bedtools2/src/utils/general/Tokenizer.cpp\n",
            "bedtools2/src/utils/general/ErrorMsg.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.cpp\n",
            "bedtools2/src/utils/general/Makefile.frag\n",
            "bedtools2/src/utils/general/BTlist.h\n",
            "bedtools2/src/utils/general/CompressionTools.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.cpp\n",
            "bedtools2/src/utils/general/InflateStreamBuf.h\n",
            "bedtools2/src/utils/general/Random.h\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.h\n",
            "bedtools2/src/utils/BlockedIntervals/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/mapping/\n",
            "bedtools2/src/utils/BamTools/include/\n",
            "bedtools2/src/utils/BamTools/.gitignore\n",
            "bedtools2/src/utils/BamTools/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/include/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools/include/SamHeader.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamReader.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/\n",
            "bedtools2/src/utils/BamTools/include/BamWriter.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.mapping.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/BamWriter.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamMultiReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/SamConstants.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAlignment.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAux.h\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.map\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.py\n",
            "bedtools2/src/utils/Contexts/ContextBase.h\n",
            "bedtools2/src/utils/Contexts/ContextClosest.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.cpp\n",
            "bedtools2/src/utils/Contexts/ContextFisher.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMerge.h\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.h\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.cpp\n",
            "bedtools2/src/utils/Contexts/ContextBase.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSummary.h\n",
            "bedtools2/src/utils/Contexts/ContextFisher.h\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.cpp\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.cpp\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.h\n",
            "bedtools2/src/utils/Contexts/ContextSample.cpp\n",
            "bedtools2/src/utils/Contexts/ContextClosest.cpp\n",
            "bedtools2/src/utils/Contexts/Makefile.frag\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSample.h\n",
            "bedtools2/src/utils/Contexts/ContextSummary.cpp\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.h\n",
            "bedtools2/src/utils/Contexts/ContextMerge.cpp\n",
            "bedtools2/src/utils/bedFile/bedFile.h\n",
            "bedtools2/src/utils/bedFile/Makefile.frag\n",
            "bedtools2/src/utils/bedFile/bedFile.cpp\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
            "bedtools2/src/utils/RecordOutputMgr/Makefile.frag\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.h\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.h\n",
            "bedtools2/src/utils/NewChromsweep/Makefile.frag\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.cpp\n",
            "bedtools2/src/utils/htslib/htslib.mk\n",
            "bedtools2/src/utils/htslib/cram/\n",
            "bedtools2/src/utils/htslib/htslib/\n",
            "bedtools2/src/utils/htslib/faidx.c\n",
            "bedtools2/src/utils/htslib/vcfutils.c\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.h\n",
            "bedtools2/src/utils/htslib/faidx.5\n",
            "bedtools2/src/utils/htslib/regidx.c\n",
            "bedtools2/src/utils/htslib/htslib.pc.in\n",
            "bedtools2/src/utils/htslib/configure.ac\n",
            "bedtools2/src/utils/htslib/INSTALL\n",
            "bedtools2/src/utils/htslib/hts_os.c\n",
            "bedtools2/src/utils/htslib/tbx.c\n",
            "bedtools2/src/utils/htslib/probaln.c\n",
            "bedtools2/src/utils/htslib/LICENSE\n",
            "bedtools2/src/utils/htslib/kfunc.c\n",
            "bedtools2/src/utils/htslib/hfile_gcs.c\n",
            "bedtools2/src/utils/htslib/errmod.c\n",
            "bedtools2/src/utils/htslib/multipart.c\n",
            "bedtools2/src/utils/htslib/hfile_internal.h\n",
            "bedtools2/src/utils/htslib/Makefile\n",
            "bedtools2/src/utils/htslib/htsfile.c\n",
            "bedtools2/src/utils/htslib/realn.c\n",
            "bedtools2/src/utils/htslib/bgzip.c\n",
            "bedtools2/src/utils/htslib/hfile_net.c\n",
            "bedtools2/src/utils/htslib/tabix.1\n",
            "bedtools2/src/utils/htslib/textutils.c\n",
            "bedtools2/src/utils/htslib/hts.c\n",
            "bedtools2/src/utils/htslib/README\n",
            "bedtools2/src/utils/htslib/vcf_sweep.c\n",
            "bedtools2/src/utils/htslib/README.md\n",
            "bedtools2/src/utils/htslib/sam.5\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.c\n",
            "bedtools2/src/utils/htslib/synced_bcf_reader.c\n",
            "bedtools2/src/utils/htslib/htslib_vars.mk\n",
            "bedtools2/src/utils/htslib/sam.c\n",
            "bedtools2/src/utils/htslib/NEWS\n",
            "bedtools2/src/utils/htslib/.gitignore\n",
            "bedtools2/src/utils/htslib/vcf.5\n",
            "bedtools2/src/utils/htslib/bgzf.c\n",
            "bedtools2/src/utils/htslib/.gitattributes\n",
            "bedtools2/src/utils/htslib/hts_internal.h\n",
            "bedtools2/src/utils/htslib/vcf.c\n",
            "bedtools2/src/utils/htslib/m4/\n",
            "bedtools2/src/utils/htslib/plugin.c\n",
            "bedtools2/src/utils/htslib/version.sh\n",
            "bedtools2/src/utils/htslib/hfile_libcurl.c\n",
            "bedtools2/src/utils/htslib/kstring.c\n",
            "bedtools2/src/utils/htslib/os/\n",
            "bedtools2/src/utils/htslib/md5.c\n",
            "bedtools2/src/utils/htslib/hfile_s3.c\n",
            "bedtools2/src/utils/htslib/htsfile.1\n",
            "bedtools2/src/utils/htslib/thread_pool.c\n",
            "bedtools2/src/utils/htslib/hfile.c\n",
            "bedtools2/src/utils/htslib/config.mk.in\n",
            "bedtools2/src/utils/htslib/tabix.c\n",
            "bedtools2/src/utils/htslib/textutils_internal.h\n",
            "bedtools2/src/utils/htslib/knetfile.c\n",
            "bedtools2/src/utils/htslib/bgzip.1\n",
            "bedtools2/src/utils/htslib/thread_pool_internal.h\n",
            "bedtools2/src/utils/htslib/os/rand.c\n",
            "bedtools2/src/utils/htslib/os/lzma_stub.h\n",
            "bedtools2/src/utils/htslib/m4/hts_prog_cc_warnings.m4\n",
            "bedtools2/src/utils/htslib/htslib/sam.h\n",
            "bedtools2/src/utils/htslib/htslib/synced_bcf_reader.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf_sweep.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf.h\n",
            "bedtools2/src/utils/htslib/htslib/khash_str2int.h\n",
            "bedtools2/src/utils/htslib/htslib/bgzf.h\n",
            "bedtools2/src/utils/htslib/htslib/cram.h\n",
            "bedtools2/src/utils/htslib/htslib/kstring.h\n",
            "bedtools2/src/utils/htslib/htslib/knetfile.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_log.h\n",
            "bedtools2/src/utils/htslib/htslib/hfile.h\n",
            "bedtools2/src/utils/htslib/htslib/thread_pool.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_defs.h\n",
            "bedtools2/src/utils/htslib/htslib/khash.h\n",
            "bedtools2/src/utils/htslib/htslib/regidx.h\n",
            "bedtools2/src/utils/htslib/htslib/faidx.h\n",
            "bedtools2/src/utils/htslib/htslib/vcfutils.h\n",
            "bedtools2/src/utils/htslib/htslib/kseq.h\n",
            "bedtools2/src/utils/htslib/htslib/kfunc.h\n",
            "bedtools2/src/utils/htslib/htslib/tbx.h\n",
            "bedtools2/src/utils/htslib/htslib/ksort.h\n",
            "bedtools2/src/utils/htslib/htslib/kbitset.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_os.h\n",
            "bedtools2/src/utils/htslib/htslib/klist.h\n",
            "bedtools2/src/utils/htslib/htslib/hts.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_endian.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.c\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.c\n",
            "bedtools2/src/utils/htslib/cram/mFILE.h\n",
            "bedtools2/src/utils/htslib/cram/cram_external.c\n",
            "bedtools2/src/utils/htslib/cram/cram.h\n",
            "bedtools2/src/utils/htslib/cram/misc.h\n",
            "bedtools2/src/utils/htslib/cram/files.c\n",
            "bedtools2/src/utils/htslib/cram/os.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/cram_io.h\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.h\n",
            "bedtools2/src/utils/htslib/cram/mFILE.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.h\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.c\n",
            "bedtools2/src/utils/htslib/cram/rANS_byte.h\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.c\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.c\n",
            "bedtools2/src/utils/htslib/cram/cram_structs.h\n",
            "bedtools2/src/utils/htslib/cram/cram_io.c\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.c\n",
            "bedtools2/src/utils/tabFile/tabFile.cpp\n",
            "bedtools2/src/utils/tabFile/tabFile.h\n",
            "bedtools2/src/utils/tabFile/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.h\n",
            "bedtools2/src/genomeCoverageBed/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.h\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "bedtools2/src/maskFastaFromBed/Makefile.frag\n",
            "bedtools2/src/regressTest/RegressTest.h\n",
            "bedtools2/src/regressTest/RegressTest.cpp\n",
            "bedtools2/src/regressTest/regressTestMain.cpp\n",
            "bedtools2/src/regressTest/Makefile.frag\n",
            "bedtools2/src/bedToIgv/bedToIgv.cpp\n",
            "bedtools2/src/bedToIgv/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.h\n",
            "bedtools2/src/nucBed/nucBedMain.cpp\n",
            "bedtools2/src/nucBed/LargeFileSupport.h\n",
            "bedtools2/src/nucBed/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.cpp\n",
            "bedtools2/src/reldist/reldist.cpp\n",
            "bedtools2/src/reldist/Makefile.frag\n",
            "bedtools2/src/reldist/reldistMain.cpp\n",
            "bedtools2/src/reldist/reldist.h\n",
            "bedtools2/src/getOverlap/getOverlap.cpp\n",
            "bedtools2/src/getOverlap/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBedMain.cpp\n",
            "bedtools2/src/randomBed/randomBed.cpp\n",
            "bedtools2/src/randomBed/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBed.h\n",
            "bedtools2/src/annotateBed/annotateBed.cpp\n",
            "bedtools2/src/annotateBed/annotateBed.h\n",
            "bedtools2/src/annotateBed/annotateMain.cpp\n",
            "bedtools2/src/annotateBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBed.h\n",
            "bedtools2/src/slopBed/slopBed.cpp\n",
            "bedtools2/src/slopBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBedMain.cpp\n",
            "bedtools2/src/mapFile/mapHelp.cpp\n",
            "bedtools2/src/mapFile/mapFile.cpp\n",
            "bedtools2/src/mapFile/Makefile.frag\n",
            "bedtools2/src/mapFile/mapFile.h\n",
            "bedtools2/src/pairToPair/pairToPair.h\n",
            "bedtools2/src/pairToPair/pairToPairMain.cpp\n",
            "bedtools2/src/pairToPair/pairToPair.cpp\n",
            "bedtools2/src/pairToPair/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.cpp\n",
            "bedtools2/src/linksBed/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.h\n",
            "bedtools2/src/linksBed/linksMain.cpp\n",
            "bedtools2/src/fisher/fisher.h\n",
            "bedtools2/src/fisher/kfunc.h\n",
            "bedtools2/src/fisher/kfunc.cpp\n",
            "bedtools2/src/fisher/Makefile.frag\n",
            "bedtools2/src/fisher/fisher.cpp\n",
            "bedtools2/src/fisher/fisherHelp.cpp\n",
            "bedtools2/src/expand/Makefile.frag\n",
            "bedtools2/src/expand/expand.cpp\n",
            "bedtools2/src/complementFile/complementHelp.cpp\n",
            "bedtools2/src/complementFile/complementFile.h\n",
            "bedtools2/src/complementFile/Makefile.frag\n",
            "bedtools2/src/complementFile/complementFile.cpp\n",
            "bedtools2/src/bedToBam/bedToBam.cpp\n",
            "bedtools2/src/bedToBam/Makefile.frag\n",
            "bedtools2/src/split/splitBed.h\n",
            "bedtools2/src/split/splitBed.cpp\n",
            "bedtools2/src/split/splitBedMain.cpp\n",
            "bedtools2/src/split/Makefile.frag\n",
            "bedtools2/src/sortBed/sortMain.cpp\n",
            "bedtools2/src/sortBed/sortBed.cpp\n",
            "bedtools2/src/sortBed/sortBed.h\n",
            "bedtools2/src/sortBed/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingHelp.cpp\n",
            "bedtools2/src/spacingFile/spacingFile.h\n",
            "bedtools2/src/spacingFile/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingFile.cpp\n",
            "bedtools2/data/gerp.chr1.bed.gz\n",
            "bedtools2/data/simpleRepeats.chr1.bed.gz\n",
            "bedtools2/data/aluY.chr1.bed.gz\n",
            "bedtools2/data/refseq.chr1.exons.bed.gz\n",
            "bedtools2/data/knownGene.hg18.chr21.short.bed\n",
            "bedtools2/data/knownGene.hg18.chr21.bed\n",
            "bedtools2/.github/workflows/\n",
            "bedtools2/.github/workflows/main.yml\n",
            "bedtools2/scripts/makeBashScripts.py\n",
            "bedtools2/scripts/vcfsort.sh\n",
            "bedtools2/genomes/human.hg38.genome\n",
            "bedtools2/genomes/mouse.mm10.genome\n",
            "bedtools2/genomes/mouse.mm8.genome\n",
            "bedtools2/genomes/human.hg18.genome\n",
            "bedtools2/genomes/human.hg19.genome\n",
            "bedtools2/genomes/mouse.mm9.genome\n",
            "bedtools2/docs/index.rst\n",
            "bedtools2/docs/Makefile\n",
            "bedtools2/docs/conf.py\n",
            "bedtools2/docs/_static/\n",
            "bedtools2/docs/content/\n",
            "bedtools2/docs/bedtools.swiss.png\n",
            "bedtools2/docs/build.sh\n",
            "bedtools2/docs/make.bat\n",
            "bedtools2/docs/bedtools.ai\n",
            "bedtools2/docs/templates/\n",
            "bedtools2/docs/themes/\n",
            "bedtools2/docs/bedtools.png\n",
            "bedtools2/docs/themes/rtd/\n",
            "bedtools2/docs/themes/rtd/theme.conf\n",
            "bedtools2/docs/themes/rtd/static/\n",
            "bedtools2/docs/themes/rtd/static/rtd.css_t\n",
            "bedtools2/docs/templates/page.html\n",
            "bedtools2/docs/templates/sidebar-intro.html\n",
            "bedtools2/docs/content/general-usage.rst\n",
            "bedtools2/docs/content/tools/\n",
            "bedtools2/docs/content/images/\n",
            "bedtools2/docs/content/quick-start.rst\n",
            "bedtools2/docs/content/example-usage.rst\n",
            "bedtools2/docs/content/overview.rst\n",
            "bedtools2/docs/content/advanced-usage.rst\n",
            "bedtools2/docs/content/history.rst\n",
            "bedtools2/docs/content/installation.rst\n",
            "bedtools2/docs/content/tips-and-tricks.rst\n",
            "bedtools2/docs/content/bedtools-suite.rst\n",
            "bedtools2/docs/content/faq.rst\n",
            "bedtools2/docs/content/related-tools.rst\n",
            "bedtools2/docs/content/images/memory-comparo.png\n",
            "bedtools2/docs/content/images/speed-comparo.png\n",
            "bedtools2/docs/content/images/fisher/\n",
            "bedtools2/docs/content/images/github-zip-button.png\n",
            "bedtools2/docs/content/images/galaxy-bedtools.png\n",
            "bedtools2/docs/content/images/bedtoolsMenu.png\n",
            "bedtools2/docs/content/images/tool-glyphs/\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/merge-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/getfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/map-speed-comparo.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-plot.png\n",
            "bedtools2/docs/content/images/tool-glyphs/subtract-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/flank-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/genomecov-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/cluster-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/complement-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/slop-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/window-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/shuffle-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/closest-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/intersect-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/barski_binding_site.png\n",
            "bedtools2/docs/content/images/fisher/res-genes.png\n",
            "bedtools2/docs/content/tools/expand.rst\n",
            "bedtools2/docs/content/tools/genomecov.rst\n",
            "bedtools2/docs/content/tools/makewindows.rst\n",
            "bedtools2/docs/content/tools/igv.rst\n",
            "bedtools2/docs/content/tools/overlap.rst\n",
            "bedtools2/docs/content/tools/nuc.rst\n",
            "bedtools2/docs/content/tools/fisher.rst\n",
            "bedtools2/docs/content/tools/cluster.rst\n",
            "bedtools2/docs/content/tools/pairtobed.rst\n",
            "bedtools2/docs/content/tools/shift.rst\n",
            "bedtools2/docs/content/tools/complement.rst\n",
            "bedtools2/docs/content/tools/bed12tobed6.rst\n",
            "bedtools2/docs/content/tools/map.rst\n",
            "bedtools2/docs/content/tools/tag.rst\n",
            "bedtools2/docs/content/tools/pairtopair.rst\n",
            "bedtools2/docs/content/tools/slop.rst\n",
            "bedtools2/docs/content/tools/sort.rst\n",
            "bedtools2/docs/content/tools/reldist.rst\n",
            "bedtools2/docs/content/tools/closest.rst\n",
            "bedtools2/docs/content/tools/bedtobam.rst\n",
            "bedtools2/docs/content/tools/maskfasta.rst\n",
            "bedtools2/docs/content/tools/bamtofastq.rst\n",
            "bedtools2/docs/content/tools/multicov.rst\n",
            "bedtools2/docs/content/tools/flank.rst\n",
            "bedtools2/docs/content/tools/random.rst\n",
            "bedtools2/docs/content/tools/merge.rst\n",
            "bedtools2/docs/content/tools/shuffle.rst\n",
            "bedtools2/docs/content/tools/window.rst\n",
            "bedtools2/docs/content/tools/bedpetobam.rst\n",
            "bedtools2/docs/content/tools/groupby.rst\n",
            "bedtools2/docs/content/tools/multiinter.rst\n",
            "bedtools2/docs/content/tools/bamtobed.rst\n",
            "bedtools2/docs/content/tools/getfasta.rst\n",
            "bedtools2/docs/content/tools/unionbedg.rst\n",
            "bedtools2/docs/content/tools/subtract.rst\n",
            "bedtools2/docs/content/tools/coverage.rst\n",
            "bedtools2/docs/content/tools/jaccard.rst\n",
            "bedtools2/docs/content/tools/links.rst\n",
            "bedtools2/docs/content/tools/annotate.rst\n",
            "bedtools2/docs/content/tools/intersect.rst\n",
            "bedtools2/docs/_static/labibi.css\n",
            "bedtools2/docs/_static/labibi.js\n",
            "bedtools2/test/split/\n",
            "bedtools2/test/bedtobam/\n",
            "bedtools2/test/cluster/\n",
            "bedtools2/test/expand/\n",
            "bedtools2/test/shuffle/\n",
            "bedtools2/test/fisher/\n",
            "bedtools2/test/complement/\n",
            "bedtools2/test/intersect/\n",
            "bedtools2/test/sort/\n",
            "bedtools2/test/subtract/\n",
            "bedtools2/test/merge/\n",
            "bedtools2/test/general/\n",
            "bedtools2/test/slop/\n",
            "bedtools2/test/reldist/\n",
            "bedtools2/test/shift/\n",
            "bedtools2/test/genomecov/\n",
            "bedtools2/test/jaccard/\n",
            "bedtools2/test/groupby/\n",
            "bedtools2/test/bamtofastq/\n",
            "bedtools2/test/bed12tobed6/\n",
            "bedtools2/test/getfasta/\n",
            "bedtools2/test/map/\n",
            "bedtools2/test/bigchroms/\n",
            "bedtools2/test/spacing/\n",
            "bedtools2/test/sample/\n",
            "bedtools2/test/flank/\n",
            "bedtools2/test/negativecontrol/\n",
            "bedtools2/test/multicov/\n",
            "bedtools2/test/closest/\n",
            "bedtools2/test/test.sh\n",
            "bedtools2/test/coverage/\n",
            "bedtools2/test/bamtobed/\n",
            "bedtools2/test/makewindows/\n",
            "bedtools2/test/makewindows/test-makewindows.sh\n",
            "bedtools2/test/makewindows/a.33bp.bed\n",
            "bedtools2/test/makewindows/a.20bp.bed\n",
            "bedtools2/test/makewindows/input.bed\n",
            "bedtools2/test/makewindows/a.19bp.bed\n",
            "bedtools2/test/bamtobed/two_blocks.sam\n",
            "bedtools2/test/bamtobed/three_blocks.sam\n",
            "bedtools2/test/bamtobed/test-bamtobed.sh\n",
            "bedtools2/test/bamtobed/numeric_tag.sam\n",
            "bedtools2/test/bamtobed/extra-long-header.sam.gz\n",
            "bedtools2/test/bamtobed/two_blocks_w_D.sam\n",
            "bedtools2/test/bamtobed/one_block.sam\n",
            "bedtools2/test/coverage/two_blocks.sam\n",
            "bedtools2/test/coverage/chr_0-100.bed\n",
            "bedtools2/test/coverage/three_blocks.bam.bai\n",
            "bedtools2/test/coverage/sam-w-del.sam\n",
            "bedtools2/test/coverage/three_blocks_nomatch.bed\n",
            "bedtools2/test/coverage/two_blocks.bam.bai\n",
            "bedtools2/test/coverage/three_blocks.sam\n",
            "bedtools2/test/coverage/one_block.bam.bai\n",
            "bedtools2/test/coverage/test-coverage.sh\n",
            "bedtools2/test/coverage/x.bed\n",
            "bedtools2/test/coverage/y.bed\n",
            "bedtools2/test/coverage/a.bed\n",
            "bedtools2/test/coverage/c.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bam\n",
            "bedtools2/test/coverage/sam-w-del.bam.bai\n",
            "bedtools2/test/coverage/b.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bed\n",
            "bedtools2/test/coverage/one_block.sam\n",
            "bedtools2/test/coverage/three_blocks_match_1bp.bed\n",
            "bedtools2/test/closest/d_d1F.bed\n",
            "bedtools2/test/closest/d_q1.bed\n",
            "bedtools2/test/closest/d_iu.bed\n",
            "bedtools2/test/closest/d_d1R.bed\n",
            "bedtools2/test/closest/d_q2.bed\n",
            "bedtools2/test/closest/tfbs.bed\n",
            "bedtools2/test/closest/null_b.bed\n",
            "bedtools2/test/closest/bug281_a.medium.bed\n",
            "bedtools2/test/closest/null_c.bed\n",
            "bedtools2/test/closest/null_a.bed\n",
            "bedtools2/test/closest/d_id.bed\n",
            "bedtools2/test/closest/a2.bed\n",
            "bedtools2/test/closest/test-closest.sh\n",
            "bedtools2/test/closest/kclosest/\n",
            "bedtools2/test/closest/close-a.bed\n",
            "bedtools2/test/closest/b.names.bed\n",
            "bedtools2/test/closest/close-b.bed\n",
            "bedtools2/test/closest/strand-test-d.bed\n",
            "bedtools2/test/closest/b-one-bp-closer.bed\n",
            "bedtools2/test/closest/islands.2.bed\n",
            "bedtools2/test/closest/bug157_b.bed\n",
            "bedtools2/test/closest/shores.bed\n",
            "bedtools2/test/closest/strand-test-a.bed\n",
            "bedtools2/test/closest/strand-test-c.bed\n",
            "bedtools2/test/closest/bug157_a.bed\n",
            "bedtools2/test/closest/strand-test-b.bed\n",
            "bedtools2/test/closest/bug281_b.bed\n",
            "bedtools2/test/closest/dmr.bed\n",
            "bedtools2/test/closest/a.bed\n",
            "bedtools2/test/closest/b.bed\n",
            "bedtools2/test/closest/a.names.bed\n",
            "bedtools2/test/closest/mdb3.bed\n",
            "bedtools2/test/closest/mdb2.bed\n",
            "bedtools2/test/closest/b2.bed\n",
            "bedtools2/test/closest/mdb1.bed\n",
            "bedtools2/test/closest/mq1.bed\n",
            "bedtools2/test/closest/d.bed\n",
            "bedtools2/test/closest/d4.bed\n",
            "bedtools2/test/closest/sortAndNaming/\n",
            "bedtools2/test/closest/islands.bed\n",
            "bedtools2/test/closest/bug281_b.flip.bed\n",
            "bedtools2/test/closest/bug281_a.flip.medium.bed\n",
            "bedtools2/test/closest/d_d2R.bed\n",
            "bedtools2/test/closest/d_d2F.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/closest/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/closest/sortAndNaming/sq1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/closest/kclosest/test-kclosest.sh\n",
            "bedtools2/test/closest/kclosest/bug471_b.bed\n",
            "bedtools2/test/closest/kclosest/bug471_a.bed\n",
            "bedtools2/test/closest/kclosest/d2.bed\n",
            "bedtools2/test/closest/kclosest/q2.bed\n",
            "bedtools2/test/closest/kclosest/d3.bed\n",
            "bedtools2/test/closest/kclosest/d1.bed\n",
            "bedtools2/test/closest/kclosest/q1.bed\n",
            "bedtools2/test/multicov/two_blocks.sam\n",
            "bedtools2/test/multicov/test-multicov.sh\n",
            "bedtools2/test/multicov/test-multi.bed\n",
            "bedtools2/test/multicov/test-multi.2.sam\n",
            "bedtools2/test/multicov/multicov.bed\n",
            "bedtools2/test/multicov/test-multi.sam\n",
            "bedtools2/test/multicov/one_block.sam\n",
            "bedtools2/test/negativecontrol/test-negativecontrol.sh\n",
            "bedtools2/test/flank/test-flank.sh\n",
            "bedtools2/test/flank/a.bed\n",
            "bedtools2/test/flank/tiny.genome\n",
            "bedtools2/test/sample/test-sample.sh\n",
            "bedtools2/test/sample/human.hg19.genome\n",
            "bedtools2/test/spacing/a.bed\n",
            "bedtools2/test/spacing/test-spacing.sh\n",
            "bedtools2/test/bigchroms/test-bigchroms.sh\n",
            "bedtools2/test/bigchroms/big4.bed\n",
            "bedtools2/test/bigchroms/make-big-chrom.py\n",
            "bedtools2/test/bigchroms/big4c.bed\n",
            "bedtools2/test/bigchroms/bigx.bed\n",
            "bedtools2/test/bigchroms/abig.bed\n",
            "bedtools2/test/map/values.bam\n",
            "bedtools2/test/map/.Rhistory\n",
            "bedtools2/test/map/a.vsorted.bed\n",
            "bedtools2/test/map/three_blocks_nomatch.bed\n",
            "bedtools2/test/map/b.vsorted.bed\n",
            "bedtools2/test/map/test.vcf\n",
            "bedtools2/test/map/values.bed\n",
            "bedtools2/test/map/test.gff\n",
            "bedtools2/test/map/values5.bed\n",
            "bedtools2/test/map/ivls.bed\n",
            "bedtools2/test/map/values4.bed\n",
            "bedtools2/test/map/t.bed\n",
            "bedtools2/test/map/ivls2.bed\n",
            "bedtools2/test/map/values3.bed\n",
            "bedtools2/test/map/.RData\n",
            "bedtools2/test/map/values2.bed\n",
            "bedtools2/test/map/d.bed\n",
            "bedtools2/test/map/genome\n",
            "bedtools2/test/map/three_blocks_match.bed\n",
            "bedtools2/test/map/test-map.sh\n",
            "bedtools2/test/map/bug262_a.bed\n",
            "bedtools2/test/map/bug262_b.bed\n",
            "bedtools2/test/map/fullFields.bam\n",
            "bedtools2/test/getfasta/test.iupac.bed\n",
            "bedtools2/test/getfasta/rna.fasta\n",
            "bedtools2/test/getfasta/t.fa\n",
            "bedtools2/test/getfasta/test.iupac.fa\n",
            "bedtools2/test/getfasta/blocks.bed\n",
            "bedtools2/test/getfasta/t_fH.fa.fai\n",
            "bedtools2/test/getfasta/t.bed\n",
            "bedtools2/test/getfasta/t.fa.fai\n",
            "bedtools2/test/getfasta/test-getfasta.sh\n",
            "bedtools2/test/getfasta/t_fH.fa\n",
            "bedtools2/test/getfasta/rna.bed\n",
            "bedtools2/test/bed12tobed6/two_blocks.bed\n",
            "bedtools2/test/bed12tobed6/bed6.bed\n",
            "bedtools2/test/bed12tobed6/three_blocks.bed\n",
            "bedtools2/test/bed12tobed6/test-bed12tobed6.sh\n",
            "bedtools2/test/bed12tobed6/one_blocks.bed\n",
            "bedtools2/test/bamtofastq/test-bamtofastq.sh\n",
            "bedtools2/test/bamtofastq/test2.sam\n",
            "bedtools2/test/bamtofastq/golden.fq\n",
            "bedtools2/test/bamtofastq/golden.fq2\n",
            "bedtools2/test/bamtofastq/test.sam\n",
            "bedtools2/test/groupby/gdc.bam\n",
            "bedtools2/test/groupby/values3_case.header.bed\n",
            "bedtools2/test/groupby/test-groupby.sh\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed\n",
            "bedtools2/test/groupby/a_vcfSVtest.vcf\n",
            "bedtools2/test/groupby/values3.7fields.header.bed\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed.2\n",
            "bedtools2/test/groupby/values3.no_header.bed\n",
            "bedtools2/test/groupby/values3.header.bed\n",
            "bedtools2/test/groupby/noPosvalues.header.bed\n",
            "bedtools2/test/groupby/test.bed\n",
            "bedtools2/test/groupby/bug569_problem.txt\n",
            "bedtools2/test/jaccard/a645.bed\n",
            "bedtools2/test/jaccard/short.bed\n",
            "bedtools2/test/jaccard/bMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bed\n",
            "bedtools2/test/jaccard/c.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bam\n",
            "bedtools2/test/jaccard/b.bed\n",
            "bedtools2/test/jaccard/b645.bed\n",
            "bedtools2/test/jaccard/e.bed\n",
            "bedtools2/test/jaccard/test-jaccard.sh\n",
            "bedtools2/test/jaccard/long.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bed\n",
            "bedtools2/test/jaccard/aMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bam\n",
            "bedtools2/test/genomecov/two_blocks.sam\n",
            "bedtools2/test/genomecov/test_ref.fa.fai\n",
            "bedtools2/test/genomecov/test-genomecov.sh\n",
            "bedtools2/test/genomecov/chip.sam\n",
            "bedtools2/test/genomecov/y.bam\n",
            "bedtools2/test/genomecov/test_ref.fa\n",
            "bedtools2/test/genomecov/sam-w-del.sam\n",
            "bedtools2/test/genomecov/three_blocks_nomatch.bed\n",
            "bedtools2/test/genomecov/three_blocks.sam\n",
            "bedtools2/test/genomecov/empty.bam\n",
            "bedtools2/test/genomecov/y.bed\n",
            "bedtools2/test/genomecov/empty.cram\n",
            "bedtools2/test/genomecov/genome.txt\n",
            "bedtools2/test/genomecov/three_blocks_match.bed\n",
            "bedtools2/test/genomecov/pair-chip.sam\n",
            "bedtools2/test/genomecov/one_block.sam\n",
            "bedtools2/test/genomecov/three_blocks_match_1bp.bed\n",
            "bedtools2/test/shift/test-shift.sh\n",
            "bedtools2/test/shift/issue_807.bed\n",
            "bedtools2/test/shift/a.bed\n",
            "bedtools2/test/shift/b.bed\n",
            "bedtools2/test/shift/tiny.genome\n",
            "bedtools2/test/shift/huge.genome\n",
            "bedtools2/test/shift/issue_807.genomesize\n",
            "bedtools2/test/reldist/issue_711.a.bed\n",
            "bedtools2/test/reldist/issue_711.b.bed\n",
            "bedtools2/test/reldist/test-reldist.sh\n",
            "bedtools2/test/slop/a.bed\n",
            "bedtools2/test/slop/tiny.genome\n",
            "bedtools2/test/slop/test-slop.sh\n",
            "bedtools2/test/general/hg19.fa.fai\n",
            "bedtools2/test/general/empty.bed\n",
            "bedtools2/test/general/a.chromheader.bed\n",
            "bedtools2/test/general/test-general.sh\n",
            "bedtools2/test/general/non-empty.bed.gz\n",
            "bedtools2/test/general/t.bed\n",
            "bedtools2/test/general/a.trackheader.bed\n",
            "bedtools2/test/merge/bamCol8Mean.txt\n",
            "bedtools2/test/merge/mergeBugTest.bed\n",
            "bedtools2/test/merge/expFormat.bed\n",
            "bedtools2/test/merge/bamCol9Mean.txt\n",
            "bedtools2/test/merge/unsorted.bed\n",
            "bedtools2/test/merge/bug254_d.bed\n",
            "bedtools2/test/merge/bug254_e.bed\n",
            "bedtools2/test/merge/bamCol6Collapse.txt\n",
            "bedtools2/test/merge/bamCol3Collapse.txt\n",
            "bedtools2/test/merge/vcfSVtest.vcf\n",
            "bedtools2/test/merge/bug254_c.bed\n",
            "bedtools2/test/merge/bamCol11Collapse.txt\n",
            "bedtools2/test/merge/a.full.bed\n",
            "bedtools2/test/merge/a.bed\n",
            "bedtools2/test/merge/test-merge.sh\n",
            "bedtools2/test/merge/precisionTest.bed\n",
            "bedtools2/test/merge/b.bed\n",
            "bedtools2/test/merge/a.names.bed\n",
            "bedtools2/test/merge/testA.vcf\n",
            "bedtools2/test/merge/bamCol1Collapse.txt\n",
            "bedtools2/test/merge/precisionTest2.bed\n",
            "bedtools2/test/merge/bamCol10Collapse.txt\n",
            "bedtools2/test/merge/bamCol5Mean.txt\n",
            "bedtools2/test/merge/a.gff\n",
            "bedtools2/test/merge/bamCol7Collapse.txt\n",
            "bedtools2/test/merge/bamCol4Mean.txt\n",
            "bedtools2/test/merge/a.full.bam\n",
            "bedtools2/test/merge/mixedStrands.bed\n",
            "bedtools2/test/merge/fullFields.bam\n",
            "bedtools2/test/subtract/test-subtract.sh\n",
            "bedtools2/test/subtract/a.bed\n",
            "bedtools2/test/subtract/c.bed\n",
            "bedtools2/test/subtract/b.bed\n",
            "bedtools2/test/subtract/b2.bed\n",
            "bedtools2/test/subtract/d.bed\n",
            "bedtools2/test/sort/names.txt\n",
            "bedtools2/test/sort/a.bed\n",
            "bedtools2/test/sort/test-sort.sh\n",
            "bedtools2/test/sort/b.bed\n",
            "bedtools2/test/intersect/two_blocks.sam\n",
            "bedtools2/test/intersect/a_with_bothUnmapped.bam\n",
            "bedtools2/test/intersect/gdc.gff\n",
            "bedtools2/test/intersect/bug223_sv1_b.vcf\n",
            "bedtools2/test/intersect/bed3.bed\n",
            "bedtools2/test/intersect/bug150_a.bed\n",
            "bedtools2/test/intersect/test_ref.fa.fai\n",
            "bedtools2/test/intersect/x.bam\n",
            "bedtools2/test/intersect/exons.issue750.bed\n",
            "bedtools2/test/intersect/performanceTest.sh\n",
            "bedtools2/test/intersect/bed6.strand.bed\n",
            "bedtools2/test/intersect/queryUnmappedMateMappedCoordsInvalid.bam\n",
            "bedtools2/test/intersect/bug223_sv1_a.vcf\n",
            "bedtools2/test/intersect/bug150_b.bed\n",
            "bedtools2/test/intersect/headerOnly.vcf\n",
            "bedtools2/test/intersect/gdc_one.bed\n",
            "bedtools2/test/intersect/bed6.strand2.bed\n",
            "bedtools2/test/intersect/null_b.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader.bed\n",
            "bedtools2/test/intersect/human.hg19.vSort.genome\n",
            "bedtools2/test/intersect/gdc.bam\n",
            "bedtools2/test/intersect/bed4.bed\n",
            "bedtools2/test/intersect/chromOrderB.bed\n",
            "bedtools2/test/intersect/bedplus.bed\n",
            "bedtools2/test/intersect/test_ref.fa\n",
            "bedtools2/test/intersect/bed5.bed\n",
            "bedtools2/test/intersect/oneRecordNoNewline.bed\n",
            "bedtools2/test/intersect/aWithHeaderVsB.txt\n",
            "bedtools2/test/intersect/null_c.bed\n",
            "bedtools2/test/intersect/j1.bed\n",
            "bedtools2/test/intersect/b.issue311.vcf\n",
            "bedtools2/test/intersect/null_a.bed\n",
            "bedtools2/test/intersect/oneUnmapped.bam\n",
            "bedtools2/test/intersect/chromOrderA.bed\n",
            "bedtools2/test/intersect/bed6.bed\n",
            "bedtools2/test/intersect/issue_773_b.bed\n",
            "bedtools2/test/intersect/new_test-intersect.sh\n",
            "bedtools2/test/intersect/chromsOutOfOrder.bed\n",
            "bedtools2/test/intersect/jim.bed\n",
            "bedtools2/test/intersect/three_blocks_nomatch.bed\n",
            "bedtools2/test/intersect/b_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/bug223_f.vcf\n",
            "bedtools2/test/intersect/a.issue311.bed\n",
            "bedtools2/test/intersect/three_blocks.sam\n",
            "bedtools2/test/intersect/gdc_exp\n",
            "bedtools2/test/intersect/bug44_b.bed\n",
            "bedtools2/test/intersect/bug223_d.vcf\n",
            "bedtools2/test/intersect/bug223_e.vcf\n",
            "bedtools2/test/intersect/a_testZeroLen.bed\n",
            "bedtools2/test/intersect/a_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/x.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/dosLineChar_a.bed\n",
            "bedtools2/test/intersect/bug44_a.vcf.gz\n",
            "bedtools2/test/intersect/b.cram\n",
            "bedtools2/test/intersect/y.bed\n",
            "bedtools2/test/intersect/test-intersect.sh\n",
            "bedtools2/test/intersect/a_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/blocks.bed12\n",
            "bedtools2/test/intersect/recordsOutOfOrder.bed\n",
            "bedtools2/test/intersect/a.bed\n",
            "bedtools2/test/intersect/jim.vcf\n",
            "bedtools2/test/intersect/c.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bam\n",
            "bedtools2/test/intersect/dosLineCharWithExtraTab_b.bed\n",
            "bedtools2/test/intersect/a.cram\n",
            "bedtools2/test/intersect/b.bed\n",
            "bedtools2/test/intersect/two_blocks_partial.bam.bai\n",
            "bedtools2/test/intersect/foo\n",
            "bedtools2/test/intersect/split.issue750.bam\n",
            "bedtools2/test/intersect/mapped_and_unmapped.sam\n",
            "bedtools2/test/intersect/two_blocks_partial.bam\n",
            "bedtools2/test/intersect/e.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_gzipped.bed.gz\n",
            "bedtools2/test/intersect/svlen.vcf\n",
            "bedtools2/test/intersect/d.bed\n",
            "bedtools2/test/intersect/splitBug155_a.bed\n",
            "bedtools2/test/intersect/multi_intersect/\n",
            "bedtools2/test/intersect/notexthdr.bam\n",
            "bedtools2/test/intersect/large_b.bed\n",
            "bedtools2/test/intersect/nonamecheck_a.bed\n",
            "bedtools2/test/intersect/issue_773_y.bed\n",
            "bedtools2/test/intersect/sortAndNaming/\n",
            "bedtools2/test/intersect/split.issue750.sam\n",
            "bedtools2/test/intersect/a_gzipped.bed.gz\n",
            "bedtools2/test/intersect/splitBug155_b.bed\n",
            "bedtools2/test/intersect/bug167_strandSweep.bed\n",
            "bedtools2/test/intersect/large_a.bed\n",
            "bedtools2/test/intersect/nonamecheck_b.bed\n",
            "bedtools2/test/intersect/b.issue311.gff\n",
            "bedtools2/test/intersect/two_blocks_partial.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bed\n",
            "bedtools2/test/intersect/human.hg19.genome\n",
            "bedtools2/test/intersect/bug187_a.bed\n",
            "bedtools2/test/intersect/strand_with_star.b.bed\n",
            "bedtools2/test/intersect/b_testZeroLen.bed\n",
            "bedtools2/test/intersect/one_block.sam\n",
            "bedtools2/test/intersect/bug187_b.bed\n",
            "bedtools2/test/intersect/three_blocks_match_1bp.bed\n",
            "bedtools2/test/intersect/strand_with_star.a.bed\n",
            "bedtools2/test/intersect/aVSb.bam\n",
            "bedtools2/test/intersect/a.bam\n",
            "bedtools2/test/intersect/sortAndNaming/num_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/intersect/sortAndNaming/exp1\n",
            "bedtools2/test/intersect/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/intersect/sortAndNaming/sq1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/\n",
            "bedtools2/test/intersect/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/q500K.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/dbchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_chr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/qchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/currChr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/db500K.bed\n",
            "bedtools2/test/intersect/multi_intersect/query3.bed\n",
            "bedtools2/test/intersect/multi_intersect/query2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d8.bed\n",
            "bedtools2/test/intersect/multi_intersect/d9.bed\n",
            "bedtools2/test/intersect/multi_intersect/query.bed\n",
            "bedtools2/test/intersect/multi_intersect/g.bed\n",
            "bedtools2/test/intersect/multi_intersect/test-multi_intersect.sh\n",
            "bedtools2/test/intersect/multi_intersect/d4.bed\n",
            "bedtools2/test/intersect/multi_intersect/d5.bed\n",
            "bedtools2/test/intersect/multi_intersect/d7.bed\n",
            "bedtools2/test/intersect/multi_intersect/d6.bed\n",
            "bedtools2/test/intersect/multi_intersect/d2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d3.bed\n",
            "bedtools2/test/intersect/multi_intersect/d1.bed\n",
            "bedtools2/test/complement/test-complement.sh\n",
            "bedtools2/test/complement/issue_503.genome\n",
            "bedtools2/test/complement/issue_503.bed\n",
            "bedtools2/test/fisher/test-fisher.sh\n",
            "bedtools2/test/fisher/sim.py\n",
            "bedtools2/test/fisher/plot.py\n",
            "bedtools2/test/fisher/cmp.sh\n",
            "bedtools2/test/fisher/dm6.fai\n",
            "bedtools2/test/fisher/t.60.genome\n",
            "bedtools2/test/fisher/get.sh\n",
            "bedtools2/test/fisher/README.md\n",
            "bedtools2/test/fisher/a.bed\n",
            "bedtools2/test/fisher/b.bed\n",
            "bedtools2/test/fisher/tumor.gff\n",
            "bedtools2/test/fisher/t.500.genome\n",
            "bedtools2/test/fisher/a_merge.bed\n",
            "bedtools2/test/fisher/shuf.sh\n",
            "bedtools2/test/fisher/test.bed\n",
            "bedtools2/test/shuffle/incl.sizedif.bed\n",
            "bedtools2/test/shuffle/test-shuffle.sh\n",
            "bedtools2/test/shuffle/excl.bed\n",
            "bedtools2/test/shuffle/simrep.bed\n",
            "bedtools2/test/shuffle/incl.bed\n",
            "bedtools2/test/expand/expand.txt\n",
            "bedtools2/test/expand/test-expand.sh\n",
            "bedtools2/test/cluster/in.bed\n",
            "bedtools2/test/cluster/test-cluster.sh\n",
            "bedtools2/test/bedtobam/test-bedtobam.sh\n",
            "bedtools2/test/bedtobam/chrsize.tmp\n",
            "bedtools2/test/split/test-split.sh\n",
            "bedtools2/test/split/randData.bed\n",
            "bedtools2/tutorial/answers.md\n",
            "bedtools2/tutorial/template.class.html\n",
            "bedtools2/tutorial/answers.html\n",
            "bedtools2/tutorial/bootstrap.css\n",
            "bedtools2/tutorial/Makefile\n",
            "bedtools2/tutorial/bedtools.html\n",
            "bedtools2/tutorial/bedtools.md\n",
            "bedtools2/tutorial/template.html\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd bedtools2; make"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2T6J7KmzRrh7",
        "outputId": "f2b4e49b-9eba-4bff-f9b5-3de03b2a3402"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Building BEDTools:\n",
            "=========================================================\n",
            "CXXFLAGS is [-g -Wall -O2 -std=c++11 ]\n",
            "DETECTED_VERSION = v2.30.0\n",
            "CURRENT_VERSION  = \n",
            "Updating version file.\n",
            "  * compiling src/bedtools.cpp\n",
            "  * compiling src/annotateBed/annotateMain.cpp\n",
            "  * compiling src/annotateBed/annotateBed.cpp\n",
            "  * compiling src/bamToBed/bamToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:88:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KuseAlignmentScore\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   88 |     bool \u001b[01;35m\u001b[KuseAlignmentScore\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastqMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtofastq_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFastq2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveFastq2\u001b[m\u001b[K    = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastq.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.cpp:15\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToBam/bedToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedtobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:60:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedpeToBam/bedpeToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedpetobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:61:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   61 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToIgv/bedToIgv.cpp\n",
            "  * compiling src/bed12ToBed6/bed12ToBed6.cpp\n",
            "  * compiling src/closestFile/closestHelp.cpp\n",
            "  * compiling src/closestFile/closestFile.cpp\n",
            "  * compiling src/clusterBed/clusterMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint cluster_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMaxDistance\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveMaxDistance\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/clusterBed/clusterBed.cpp\n",
            "  * compiling src/complementFile/complementHelp.cpp\n",
            "  * compiling src/complementFile/complementFile.cpp\n",
            "  * compiling src/coverageFile/coverageHelp.cpp\n",
            "  * compiling src/coverageFile/coverageFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvirtual void CoverageFile::giveFinalReport(RecordOutputMgr*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:100:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  100 |                 \u001b[01;35m\u001b[Kasprintf(&depthPctString, \"%0.7f\", depthPct)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doMean(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:181:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  181 |         \u001b[01;35m\u001b[Kasprintf(&meanString, \"%0.7f\", mean)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doHist(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:210:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  210 |                 \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doDefault(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:230:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  230 |         \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/expand/expand.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBedMain.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBed.cpp\n",
            "  * compiling src/flankBed/flankBedMain.cpp\n",
            "  * compiling src/flankBed/flankBed.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/GenomeFile/NewGenomeFile.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/getOverlap/getOverlap.cpp\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint getoverlap_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveColumns\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveColumns\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/groupBy/groupByHelp.cpp\n",
            "  * compiling src/groupBy/groupBy.cpp\n",
            "  * compiling src/intersectFile/intersectHelp.cpp\n",
            "  * compiling src/intersectFile/intersectFile.cpp\n",
            "  * compiling src/fisher/fisherHelp.cpp\n",
            "  * compiling src/fisher/fisher.cpp\n",
            "  * compiling src/fisher/kfunc.cpp\n",
            "  * compiling src/jaccard/jaccardHelp.cpp\n",
            "  * compiling src/jaccard/jaccard.cpp\n",
            "  * compiling src/linksBed/linksMain.cpp\n",
            "  * compiling src/linksBed/linksBed.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "  * compiling src/mapFile/mapHelp.cpp\n",
            "  * compiling src/mapFile/mapFile.cpp\n",
            "  * compiling src/mergeFile/mergeHelp.cpp\n",
            "  * compiling src/mergeFile/mergeFile.cpp\n",
            "  * compiling src/multiBamCov/multiBamCovMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multibamcov_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBed\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveBed\u001b[m\u001b[K           = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:39:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBams\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   39 |     bool \u001b[01;35m\u001b[KhaveBams\u001b[m\u001b[K          = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiBamCov/multiBamCov.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/BamReader.hpp:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamMultiReader.h:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.h:16\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multiintersect_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:45:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   45 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBed.cpp\n",
            "  * compiling src/nucBed/nucBedMain.cpp\n",
            "  * compiling src/nucBed/nucBed.cpp\n",
            "  * compiling src/pairToBed/pairToBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:43:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   43 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToBed/pairToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid BedIntersectPE::FindSpanningOverlaps(const BEDPE&, std::vector<BED>&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:257:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  257 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BedIntersectPE::FindOneOrMoreSpanningOverlaps(const BEDPE&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:310:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  310 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPairMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtopair_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPair.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid PairToPair::FindOverlaps(const BEDPE&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:112:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound1\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  112 |         bool \u001b[01;35m\u001b[Kfound1\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:113:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  113 |         bool \u001b[01;35m\u001b[Kfound2\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/randomBed/randomBedMain.cpp\n",
            "  * compiling src/randomBed/randomBed.cpp\n",
            "  * compiling src/regressTest/regressTestMain.cpp\n",
            "  * compiling src/regressTest/RegressTest.cpp\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::executeAndCompareCorrectness(const fileListType&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:431:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  431 |         \u001b[01;35m\u001b[Ksystem(diffCmd.c_str())\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::startMemoryProfile(bool)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:573:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  573 |         \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::calcMemoryStats()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:622:22:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  622 |                 \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/reldist/reldistMain.cpp\n",
            "  * compiling src/reldist/reldist.cpp\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RelativeDistance::CalculateRelativeDistance()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:129:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[KCHRPOS\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  129 |             if (\u001b[01;35m\u001b[Klow_idx != chrom_mids->size() - 1\u001b[m\u001b[K)\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/sampleFile/sampleHelp.cpp\n",
            "  * compiling src/sampleFile/sampleFile.cpp\n",
            "  * compiling src/shiftBed/shiftBedMain.cpp\n",
            "  * compiling src/shiftBed/shiftBed.cpp\n",
            "  * compiling src/shuffleBed/shuffleBedMain.cpp\n",
            "  * compiling src/shuffleBed/shuffleBed.cpp\n",
            "  * compiling src/slopBed/slopBedMain.cpp\n",
            "  * compiling src/slopBed/slopBed.cpp\n",
            "  * compiling src/sortBed/sortMain.cpp\n",
            "  * compiling src/sortBed/sortBed.cpp\n",
            "  * compiling src/spacingFile/spacingHelp.cpp\n",
            "  * compiling src/spacingFile/spacingFile.cpp\n",
            "  * compiling src/split/splitBedMain.cpp\n",
            "  * compiling src/split/splitBed.cpp\n",
            "  * compiling src/subtractFile/subtractHelp.cpp\n",
            "  * compiling src/subtractFile/subtractFile.cpp\n",
            "  * compiling src/summaryFile/summaryHelp.cpp\n",
            "  * compiling src/summaryFile/summaryFile.cpp\n",
            "  * compiling src/tagBam/tagBamMain.cpp\n",
            "  * compiling src/tagBam/tagBam.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.h:17\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint unionbedgraphs_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:46:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   46 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:49:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   49 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphs.cpp\n",
            "  * compiling src/windowBed/windowMain.cpp\n",
            "  * compiling src/windowBed/windowBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/windowMaker/windowMakerMain.cpp\n",
            "  * compiling src/windowMaker/windowMaker.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BufferedStreamMgr::readFileChunk()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:148:37:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  148 |                 if (\u001b[01;35m\u001b[K_mainBufCurrLen < _useBufSize\u001b[m\u001b[K) {\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~^~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.h:14\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp:8\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/Record.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint BlockMgr::getNonRedundantOverlap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:194:35:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  194 |                 for (int i = 1; \u001b[01;35m\u001b[Ki < _overlapStarts.size()\u001b[m\u001b[K; ++i)\n",
            "      |                                 \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "  * compiling src/utils/bedFile/bedFile.cpp\n",
            "  * compiling src/utils/BinTree/BinTree.cpp\n",
            "  * compiling src/utils/version/version.cpp\n",
            "  * compiling src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "  * compiling src/utils/chromsweep/chromsweep.cpp\n",
            "  * compiling src/utils/Contexts/ContextBase.cpp\n",
            "  * compiling src/utils/Contexts/ContextIntersect.cpp\n",
            "  * compiling src/utils/Contexts/ContextFisher.cpp\n",
            "  * compiling src/utils/Contexts/ContextMap.cpp\n",
            "  * compiling src/utils/Contexts/ContextSample.cpp\n",
            "  * compiling src/utils/Contexts/ContextSpacing.cpp\n",
            "  * compiling src/utils/Contexts/ContextMerge.cpp\n",
            "  * compiling src/utils/Contexts/ContextJaccard.cpp\n",
            "  * compiling src/utils/Contexts/ContextClosest.cpp\n",
            "  * compiling src/utils/Contexts/ContextSubtract.cpp\n",
            "  * compiling src/utils/Contexts/ContextCoverage.cpp\n",
            "  * compiling src/utils/Contexts/ContextComplement.cpp\n",
            "  * compiling src/utils/Contexts/ContextGroupBy.cpp\n",
            "  * compiling src/utils/Contexts/ContextSummary.cpp\n",
            "  * compiling src/utils/general/ParseTools.cpp\n",
            "  * compiling src/utils/general/PushBackStreamBuf.cpp\n",
            "  * compiling src/utils/general/CompressionTools.cpp\n",
            "  * compiling src/utils/general/Tokenizer.cpp\n",
            "  * compiling src/utils/general/CommonHelp.cpp\n",
            "  * compiling src/utils/general/ErrorMsg.cpp\n",
            "  * compiling src/utils/general/Random.cpp\n",
            "g++ -g -Wall -O2 -std=c++11  -Isrc/utils/gzstream  -c -o obj/gzstream.o src/utils/gzstream/gzstream.C\n",
            "  * compiling src/utils/fileType/fileType.cpp\n",
            "  * compiling src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool FileRecordTypeChecker::isTextDelimtedFormat(const char*, size_t)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:366:33:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  366 |                 for (int j=0; \u001b[01;35m\u001b[Kj < len\u001b[m\u001b[K; j++) {\n",
            "      |                               \u001b[01;35m\u001b[K~~^~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/bedFilePE/bedFilePE.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOps.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "  * compiling src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "  * compiling src/utils/NewChromsweep/CloseSweep.cpp\n",
            "  * compiling src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "  * compiling src/utils/tabFile/tabFile.cpp\n",
            "  * compiling src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "  * compiling src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "  * compiling src/utils/Fasta/Fasta.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSequence(std::string)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:292:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  292 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSubSequence(std::string, CHRPOS, CHRPOS)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:334:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  334 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), (off_t) seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/Fasta/split.cpp\n",
            "  * compiling src/utils/VectorOps/VectorOps.cpp\n",
            "  * compiling src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "  * compiling src/utils/GenomeFile/GenomeFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid GenomeFile::loadGenomeFileIntoMap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:60:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kc2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |                     long \u001b[01;35m\u001b[Kc2\u001b[m\u001b[K;\n",
            "      |                          \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "  * compiling src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::printClosest(RecordKeyVector&, const std::vector<long int>*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:170:56:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  170 |                 if (\u001b[01;35m\u001b[Kcontext->getNumClosestHitsWanted() > keyList.size()\u001b[m\u001b[K)\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::reportOverlapSummary(RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:455:38:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  455 |                 for (size_t i = 1; \u001b[01;35m\u001b[Ki < _context->getNumInputFiles()\u001b[m\u001b[K; i++)\n",
            "      |                                    \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/ToolBase/ToolBase.cpp\n",
            "  * compiling src/utils/driver/BedtoolsDriver.cpp\n",
            "cd src/utils/htslib && make config.h\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_FSEEKO 1' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "cd src/utils/htslib && make lib-static\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"1.9\"' > version.h\n",
            "gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n",
            "\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Khts.c:1397:59:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1397 |     idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n",
            "      |                                                           \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:53\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2095:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3748:13\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3823:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:4126:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/string_alloc.c:41\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:151:5\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:143:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:143:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  143 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n",
            "      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "- Building main bedtools binary.\n",
            "done.\n",
            "- Creating executables for old CLI.\n",
            "done.\n",
            "  * compiling src/htsutil.cpp\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://github.com/bedops/bedops/releases/download/v2.4.41/bedops_linux_x86_64-v2.4.41.tar.bz2\n",
        "! tar jxvf bedops_linux_x86_64-v2.4.41.tar.bz2\n",
        "! cp bin/* /usr/local/bin"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pFViLavtpoMM",
        "outputId": "d62543cc-65af-4f59-cb4e-acfbd88ba9cb"
      },
      "execution_count": 69,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-12 22:51:53--  https://github.com/bedops/bedops/releases/download/v2.4.41/bedops_linux_x86_64-v2.4.41.tar.bz2\n",
            "Resolving github.com (github.com)... 140.82.112.3\n",
            "Connecting to github.com (github.com)|140.82.112.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/12932856/7baae005-767f-4700-bd69-68f44f9a01bf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240612%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240612T225153Z&X-Amz-Expires=300&X-Amz-Signature=b876549b4edc6b141000256e9cb3bca62d00f96a296e553af0fcd9de26b04264&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=12932856&response-content-disposition=attachment%3B%20filename%3Dbedops_linux_x86_64-v2.4.41.tar.bz2&response-content-type=application%2Foctet-stream [following]\n",
            "--2024-06-12 22:51:53--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/12932856/7baae005-767f-4700-bd69-68f44f9a01bf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240612%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240612T225153Z&X-Amz-Expires=300&X-Amz-Signature=b876549b4edc6b141000256e9cb3bca62d00f96a296e553af0fcd9de26b04264&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=12932856&response-content-disposition=attachment%3B%20filename%3Dbedops_linux_x86_64-v2.4.41.tar.bz2&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 26814112 (26M) [application/octet-stream]\n",
            "Saving to: ‘bedops_linux_x86_64-v2.4.41.tar.bz2’\n",
            "\n",
            "bedops_linux_x86_64 100%[===================>]  25.57M   167MB/s    in 0.2s    \n",
            "\n",
            "2024-06-12 22:51:54 (167 MB/s) - ‘bedops_linux_x86_64-v2.4.41.tar.bz2’ saved [26814112/26814112]\n",
            "\n",
            "bin/\n",
            "bin/bam2bed-typical\n",
            "bin/bam2bed-megarow\n",
            "bin/bam2bed-float128\n",
            "bin/bam2bed_gnuParallel-typical\n",
            "bin/bam2bed_gnuParallel-megarow\n",
            "bin/bam2bed_gnuParallel-float128\n",
            "bin/bam2bed_sge-typical\n",
            "bin/bam2bed_sge-megarow\n",
            "bin/bam2bed_sge-float128\n",
            "bin/bam2bed_slurm-typical\n",
            "bin/bam2bed_slurm-megarow\n",
            "bin/bam2bed_slurm-float128\n",
            "bin/bam2starch-typical\n",
            "bin/bam2starch-megarow\n",
            "bin/bam2starch-float128\n",
            "bin/bam2starch_gnuParallel-typical\n",
            "bin/bam2starch_gnuParallel-megarow\n",
            "bin/bam2starch_gnuParallel-float128\n",
            "bin/bam2starch_sge-typical\n",
            "bin/bam2starch_sge-megarow\n",
            "bin/bam2starch_sge-float128\n",
            "bin/bam2starch_slurm-typical\n",
            "bin/bam2starch_slurm-megarow\n",
            "bin/bam2starch_slurm-float128\n",
            "bin/gff2bed-typical\n",
            "bin/gff2bed-megarow\n",
            "bin/gff2bed-float128\n",
            "bin/gff2starch-typical\n",
            "bin/gff2starch-megarow\n",
            "bin/gff2starch-float128\n",
            "bin/gtf2bed-typical\n",
            "bin/gtf2bed-megarow\n",
            "bin/gtf2bed-float128\n",
            "bin/gtf2starch-typical\n",
            "bin/gtf2starch-megarow\n",
            "bin/gtf2starch-float128\n",
            "bin/gvf2bed-typical\n",
            "bin/gvf2bed-megarow\n",
            "bin/gvf2bed-float128\n",
            "bin/gvf2starch-typical\n",
            "bin/gvf2starch-megarow\n",
            "bin/gvf2starch-float128\n",
            "bin/psl2bed-typical\n",
            "bin/psl2bed-megarow\n",
            "bin/psl2bed-float128\n",
            "bin/psl2starch-typical\n",
            "bin/psl2starch-megarow\n",
            "bin/psl2starch-float128\n",
            "bin/rmsk2bed-typical\n",
            "bin/rmsk2bed-megarow\n",
            "bin/rmsk2bed-float128\n",
            "bin/rmsk2starch-typical\n",
            "bin/rmsk2starch-megarow\n",
            "bin/rmsk2starch-float128\n",
            "bin/sam2bed-typical\n",
            "bin/sam2bed-megarow\n",
            "bin/sam2bed-float128\n",
            "bin/sam2starch-typical\n",
            "bin/sam2starch-megarow\n",
            "bin/sam2starch-float128\n",
            "bin/vcf2bed-typical\n",
            "bin/vcf2bed-megarow\n",
            "bin/vcf2bed-float128\n",
            "bin/vcf2starch-typical\n",
            "bin/vcf2starch-megarow\n",
            "bin/vcf2starch-float128\n",
            "bin/wig2bed-typical\n",
            "bin/wig2bed-megarow\n",
            "bin/wig2bed-float128\n",
            "bin/wig2starch-typical\n",
            "bin/wig2starch-megarow\n",
            "bin/wig2starch-float128\n",
            "bin/starchcluster_sge-typical\n",
            "bin/starchcluster_gnuParallel-typical\n",
            "bin/starchcluster_slurm-typical\n",
            "bin/starch-diff-typical\n",
            "bin/starchcluster_sge-megarow\n",
            "bin/starchcluster_gnuParallel-megarow\n",
            "bin/starchcluster_slurm-megarow\n",
            "bin/starch-diff-megarow\n",
            "bin/starchcluster_sge-float128\n",
            "bin/starchcluster_gnuParallel-float128\n",
            "bin/starchcluster_slurm-float128\n",
            "bin/starch-diff-float128\n",
            "bin/sort-bed-typical\n",
            "bin/update-sort-bed-slurm-typical\n",
            "bin/update-sort-bed-starch-slurm-typical\n",
            "bin/update-sort-bed-migrate-candidates-typical\n",
            "bin/bedops-typical\n",
            "bin/closest-features-typical\n",
            "bin/bedmap-typical\n",
            "bin/bedextract-typical\n",
            "bin/starch-typical\n",
            "bin/unstarch-typical\n",
            "bin/starchcat-typical\n",
            "bin/starchstrip-typical\n",
            "bin/convert2bed-typical\n",
            "bin/sort-bed-megarow\n",
            "bin/update-sort-bed-slurm-megarow\n",
            "bin/update-sort-bed-starch-slurm-megarow\n",
            "bin/update-sort-bed-migrate-candidates-megarow\n",
            "bin/bedops-megarow\n",
            "bin/closest-features-megarow\n",
            "bin/bedmap-megarow\n",
            "bin/bedextract-megarow\n",
            "bin/starch-megarow\n",
            "bin/unstarch-megarow\n",
            "bin/starchcat-megarow\n",
            "bin/starchstrip-megarow\n",
            "bin/convert2bed-megarow\n",
            "bin/sort-bed-float128\n",
            "bin/update-sort-bed-slurm-float128\n",
            "bin/update-sort-bed-starch-slurm-float128\n",
            "bin/update-sort-bed-migrate-candidates-float128\n",
            "bin/bedops-float128\n",
            "bin/closest-features-float128\n",
            "bin/bedmap-float128\n",
            "bin/bedextract-float128\n",
            "bin/starch-float128\n",
            "bin/unstarch-float128\n",
            "bin/starchcat-float128\n",
            "bin/starchstrip-float128\n",
            "bin/convert2bed-float128\n",
            "bin/switch-BEDOPS-binary-type\n",
            "bin/bam2bed\n",
            "bin/bam2bed_gnuParallel\n",
            "bin/bam2bed_sge\n",
            "bin/bam2bed_slurm\n",
            "bin/bam2starch\n",
            "bin/bam2starch_gnuParallel\n",
            "bin/bam2starch_sge\n",
            "bin/bam2starch_slurm\n",
            "bin/gff2bed\n",
            "bin/gff2starch\n",
            "bin/gtf2bed\n",
            "bin/gtf2starch\n",
            "bin/gvf2bed\n",
            "bin/gvf2starch\n",
            "bin/psl2bed\n",
            "bin/psl2starch\n",
            "bin/rmsk2bed\n",
            "bin/rmsk2starch\n",
            "bin/sam2bed\n",
            "bin/sam2starch\n",
            "bin/vcf2bed\n",
            "bin/vcf2starch\n",
            "bin/wig2bed\n",
            "bin/wig2starch\n",
            "bin/starchcluster_sge\n",
            "bin/starchcluster_gnuParallel\n",
            "bin/starchcluster_slurm\n",
            "bin/starch-diff\n",
            "bin/sort-bed\n",
            "bin/update-sort-bed-slurm\n",
            "bin/update-sort-bed-starch-slurm\n",
            "bin/update-sort-bed-migrate-candidates\n",
            "bin/bedops\n",
            "bin/closest-features\n",
            "bin/bedmap\n",
            "bin/bedextract\n",
            "bin/starch\n",
            "bin/unstarch\n",
            "bin/starchcat\n",
            "bin/starchstrip\n",
            "bin/convert2bed\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/bedtools2/bin/* /usr/local/bin/"
      ],
      "metadata": {
        "id": "wU_WAqueRtJG"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "families = [\"AAA\", \"Helicase_C\", \"Metallophos\", \"Methyltransf_11\"]"
      ],
      "metadata": {
        "id": "mZj0FfC3Yih_"
      },
      "execution_count": 54,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install biopython"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KQTOaDliZdIo",
        "outputId": "7f30c27a-66b6-4918-dbef-26d01240bad5"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting biopython\n",
            "  Downloading biopython-1.83-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m21.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython) (1.25.2)\n",
            "Installing collected packages: biopython\n",
            "Successfully installed biopython-1.83\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "from collections import defaultdict\n",
        "from Bio import SearchIO\n",
        "\n",
        "def get_hmmer_data(filename):\n",
        "  attribs = ['accession', 'bias', 'bitscore', 'description', 'cluster_num', 'domain_exp_num',  'domain_included_num', 'domain_obs_num', 'domain_reported_num', 'env_num', 'evalue', 'id', 'query_id', 'overlap_num', 'region_num']\n",
        "  hits = defaultdict(list)\n",
        "\n",
        "  with open(filename) as handle:\n",
        "      for queryresult in SearchIO.parse(handle, 'hmmer3-tab'):\n",
        "        for hit in queryresult.hits:\n",
        "          for attrib in attribs:\n",
        "            hits[attrib].append(getattr(hit, attrib))\n",
        "\n",
        "  return pd.DataFrame.from_dict(hits)\n"
      ],
      "metadata": {
        "id": "ucWVo0giZMqn"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"cuted.fa\", \"w\")\n",
        "with open(\"genes5_pqs.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "R90FB56tWNCE"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Angomonas_deanei"
      ],
      "metadata": {
        "id": "eC0LxgzuWHwg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "rO5D_8KySxmp"
      },
      "execution_count": 150,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "s6orHPzhJTOE"
      },
      "outputs": [],
      "source": [
        "!bedtools getfasta -fi GCA_903995115.1_Adeanei_nanopore_chromosomes_genomic.fna -bed Angomonas_deanei_genes_pqs.bed -fo genes_orthol.fa"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Angomonas_deanei_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 930
        },
        "id": "UVjX2CRdYDNm",
        "outputId": "3002086c-5f00-4d6d-d68d-11133b021b2b"
      },
      "execution_count": 151,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   1.3     295.7   \n",
              "1            -   5.2     290.1   \n",
              "2            -   0.4      13.0   \n",
              "3            -   0.0      12.5   \n",
              "4            -   0.0     152.1   \n",
              "...        ...   ...       ...   \n",
              "4512         -   3.3      34.8   \n",
              "4513         -   7.6      24.1   \n",
              "4514         -  11.5      12.0   \n",
              "4515         -   2.9      11.2   \n",
              "4516         -  13.5       9.3   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0           14-3-3 protein, putative [Angomonas deanei]            0   \n",
              "1           14-3-3 protein, putative [Angomonas deanei]            0   \n",
              "2     hypothetical protein, conserved [Angomonas dea...            0   \n",
              "3     D-isomer specific 2-hydroxyacid dehydrogenase,...            0   \n",
              "4     D-isomer specific 2-hydroxyacid dehydrogenase,...            0   \n",
              "...                                                 ...          ...   \n",
              "4512  Zinc finger, ZZ type, putative [Angomonas deanei]            1   \n",
              "4513  C1 domain/Zinc finger, ZZ type/HECT-domain (ub...            0   \n",
              "4514  SPRY domain containing protein, putative [Ango...            0   \n",
              "4515  ABC transporter transmembrane region containin...            0   \n",
              "4516  FYVE zinc finger containing protein, putative ...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                1.7                    0               1   \n",
              "3                1.3                    0               1   \n",
              "4                1.2                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "4512             2.7                    1               2   \n",
              "4513             1.4                    1               1   \n",
              "4514             1.9                    0               1   \n",
              "4515             1.4                    0               1   \n",
              "4516             2.6                    0               2   \n",
              "\n",
              "      domain_reported_num  env_num        evalue            id      query_id  \\\n",
              "0                       1        1  1.100000e-88  CAD2212839.1        14-3-3   \n",
              "1                       1        1  5.600000e-87  CAD2222833.1        14-3-3   \n",
              "2                       1        1  3.200000e-02  CAD2222143.1        14-3-3   \n",
              "3                       1        1  4.900000e-02  CAD2222995.1    2-Hacid_dh   \n",
              "4                       1        1  5.400000e-45  CAD2218845.1  2-Hacid_dh_C   \n",
              "...                   ...      ...           ...           ...           ...   \n",
              "4512                    2        2  6.300000e-09  CAD2215290.1            ZZ   \n",
              "4513                    1        1  1.400000e-05  CAD2222144.1            ZZ   \n",
              "4514                    1        1  7.900000e-02  CAD2212816.1            ZZ   \n",
              "4515                    1        1  1.400000e-01  CAD2219288.1            ZZ   \n",
              "4516                    2        2  5.600000e-01  CAD2222273.1            ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           1  \n",
              "3               0           1  \n",
              "4               0           1  \n",
              "...           ...         ...  \n",
              "4512            0           2  \n",
              "4513            0           1  \n",
              "4514            0           1  \n",
              "4515            0           1  \n",
              "4516            0           2  \n",
              "\n",
              "[4517 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-a924d8fe-ad99-4729-99e4-3d2146aa2cd4\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>1.3</td>\n",
              "      <td>295.7</td>\n",
              "      <td>14-3-3 protein, putative [Angomonas deanei]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.100000e-88</td>\n",
              "      <td>CAD2212839.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>5.2</td>\n",
              "      <td>290.1</td>\n",
              "      <td>14-3-3 protein, putative [Angomonas deanei]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.600000e-87</td>\n",
              "      <td>CAD2222833.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.4</td>\n",
              "      <td>13.0</td>\n",
              "      <td>hypothetical protein, conserved [Angomonas dea...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.7</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.200000e-02</td>\n",
              "      <td>CAD2222143.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>12.5</td>\n",
              "      <td>D-isomer specific 2-hydroxyacid dehydrogenase,...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.900000e-02</td>\n",
              "      <td>CAD2222995.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>152.1</td>\n",
              "      <td>D-isomer specific 2-hydroxyacid dehydrogenase,...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.400000e-45</td>\n",
              "      <td>CAD2218845.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4512</th>\n",
              "      <td>-</td>\n",
              "      <td>3.3</td>\n",
              "      <td>34.8</td>\n",
              "      <td>Zinc finger, ZZ type, putative [Angomonas deanei]</td>\n",
              "      <td>1</td>\n",
              "      <td>2.7</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>6.300000e-09</td>\n",
              "      <td>CAD2215290.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4513</th>\n",
              "      <td>-</td>\n",
              "      <td>7.6</td>\n",
              "      <td>24.1</td>\n",
              "      <td>C1 domain/Zinc finger, ZZ type/HECT-domain (ub...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.400000e-05</td>\n",
              "      <td>CAD2222144.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4514</th>\n",
              "      <td>-</td>\n",
              "      <td>11.5</td>\n",
              "      <td>12.0</td>\n",
              "      <td>SPRY domain containing protein, putative [Ango...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.9</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>7.900000e-02</td>\n",
              "      <td>CAD2212816.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4515</th>\n",
              "      <td>-</td>\n",
              "      <td>2.9</td>\n",
              "      <td>11.2</td>\n",
              "      <td>ABC transporter transmembrane region containin...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.400000e-01</td>\n",
              "      <td>CAD2219288.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4516</th>\n",
              "      <td>-</td>\n",
              "      <td>13.5</td>\n",
              "      <td>9.3</td>\n",
              "      <td>FYVE zinc finger containing protein, putative ...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.6</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>5.600000e-01</td>\n",
              "      <td>CAD2222273.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>4517 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a924d8fe-ad99-4729-99e4-3d2146aa2cd4')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-a924d8fe-ad99-4729-99e4-3d2146aa2cd4 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-a924d8fe-ad99-4729-99e4-3d2146aa2cd4');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-ce089dfe-7423-41a7-ad1f-35b3683be501\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ce089dfe-7423-41a7-ad1f-35b3683be501')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-ce089dfe-7423-41a7-ad1f-35b3683be501 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 4517,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 10.064247026118396,\n        \"min\": 0.0,\n        \"max\": 201.6,\n        \"num_unique_values\": 360,\n        \"samples\": [\n          17.2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 62.853351655739104,\n        \"min\": 3.9,\n        \"max\": 873.6,\n        \"num_unique_values\": 1303,\n        \"samples\": [\n          55.2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 895,\n        \"samples\": [\n          \"Oxygenase domain of the 2OGFeDO superfamily/Thymine dioxygenase JBP1 DNA-binding domain containing protein, putative [Angomonas deanei]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 6,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.773675155396217,\n        \"min\": 1.0,\n        \"max\": 33.9,\n        \"num_unique_values\": 158,\n        \"samples\": [\n          12.3\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 22,\n        \"num_unique_values\": 19,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 40,\n        \"num_unique_values\": 28,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          10\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 40,\n        \"num_unique_values\": 28,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.6526324453116686,\n        \"min\": 3.2e-263,\n        \"max\": 9.9,\n        \"num_unique_values\": 1976,\n        \"samples\": [\n          9.2e-05\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2423,\n        \"samples\": [\n          \"CAD2221233.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 339,\n        \"samples\": [\n          \"UBA\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 16,\n        \"num_unique_values\": 15,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 40,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 151
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"angomonas_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "d9NFeYE8b3Rp"
      },
      "execution_count": 152,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_903995115.1_Adeanei_nanopore_chromosomes_genomic.fna -bed angomonas_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "id": "c_vc9jwahiRA"
      },
      "execution_count": 66,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "d6T9hxCZjmli"
      },
      "execution_count": 67,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import random\n",
        "f = open(\"Angomonas_filtered_genes.bed\", \"w\")\n",
        "count_hmmer_genes = 0\n",
        "\n",
        "while count_hmmer_genes != 4:\n",
        "  index = random.randint(0, len(hmmer_data))\n",
        "  protein = hmmer_data.iloc[[index]]\n",
        "  gene = genome.loc[genome[\"attributes\"].str.contains(protein[\"id\"].item())]\n",
        "  while gene[\"type\"].item() != \"gene\":\n",
        "    gene = genome.iloc[[gene.index.item() - 1]]\n",
        "  if not ((gene[\"end\"].item() - gene[\"start\"].item()) <= 2000):\n",
        "    continue\n",
        "  f.write(f'{gene.iloc[[0]][\"chrom\"].item()}\\t{gene.iloc[[0]][\"start\"].item()}\\t{gene.iloc[[0]][\"end\"].item()}\\t{protein[\"id\"].item()}\\n')\n",
        "  count_hmmer_genes += 1\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "Ev2dQYaGSTG-"
      },
      "execution_count": 153,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_903995115.1_Adeanei_nanopore_chromosomes_genomic.fna -bed Angomonas_filtered_genes.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "id": "k7oZkjMFTYGD"
      },
      "execution_count": 154,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "leishmania_panamensi_filtered"
      ],
      "metadata": {
        "id": "sNeWlZcLl7SR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "qLhdplIFprcd"
      },
      "execution_count": 160,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"leishmania_panamensi_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 947
        },
        "id": "EIqJV_KKl9jU",
        "outputId": "e80ee7e6-089f-43c6-97aa-eb4cb28a5228"
      },
      "execution_count": 161,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   0.3     286.6   \n",
              "1            -   1.1     283.2   \n",
              "2            -   0.2      12.3   \n",
              "3            -   0.1     121.6   \n",
              "4            -   0.1     181.5   \n",
              "...        ...   ...       ...   \n",
              "3632         -   8.1      21.4   \n",
              "3633         -   9.7      18.8   \n",
              "3634         -   3.4      12.6   \n",
              "3635         -  22.8      11.2   \n",
              "3636         -   9.0      11.2   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0     14-3-3 protein-like protein [Leishmania paname...            0   \n",
              "1      14-3-3 protein, putative [Leishmania panamensis]            0   \n",
              "2     hypothetical protein LPMP_180840 [Leishmania p...            0   \n",
              "3     D-3-phosphoglycerate dehydrogenase-like protei...            0   \n",
              "4     D-3-phosphoglycerate dehydrogenase-like protei...            0   \n",
              "...                                                 ...          ...   \n",
              "3632  hypothetical protein LPMP_321160 [Leishmania p...            0   \n",
              "3633  hypothetical protein LPMP_324060 [Leishmania p...            0   \n",
              "3634  dual specificity protein phosphatase, putative...            0   \n",
              "3635   protein kinase, putative [Leishmania panamensis]            0   \n",
              "3636  ubiquitin-protein ligase-like, putative [Leish...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                2.8                    0               3   \n",
              "3                1.2                    1               1   \n",
              "4                1.3                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "3632             2.2                    1               2   \n",
              "3633             1.7                    1               1   \n",
              "3634             1.8                    0               1   \n",
              "3635             2.7                    0               2   \n",
              "3636             1.5                    0               1   \n",
              "\n",
              "      domain_reported_num  env_num        evalue          id      query_id  \\\n",
              "0                       1        1  5.000000e-86  AIO02465.1        14-3-3   \n",
              "1                       1        1  5.400000e-85  AIN96231.1        14-3-3   \n",
              "2                       3        3  3.900000e-02  AIN97265.1        14-3-3   \n",
              "3                       1        1  7.200000e-36  AIN95237.1    2-Hacid_dh   \n",
              "4                       1        1  3.900000e-54  AIN95237.1  2-Hacid_dh_C   \n",
              "...                   ...      ...           ...         ...           ...   \n",
              "3632                    2        2  7.000000e-05  AIO01002.1            ZZ   \n",
              "3633                    1        1  4.600000e-04  AIO01287.1            ZZ   \n",
              "3634                    1        1  4.000000e-02  AIN97681.1            ZZ   \n",
              "3635                    2        2  1.000000e-01  AIN98802.1            ZZ   \n",
              "3636                    1        1  1.100000e-01  AIN95717.1            ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           3  \n",
              "3               0           1  \n",
              "4               0           1  \n",
              "...           ...         ...  \n",
              "3632            0           2  \n",
              "3633            0           1  \n",
              "3634            0           1  \n",
              "3635            0           2  \n",
              "3636            0           1  \n",
              "\n",
              "[3637 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-eb7892d4-b8f6-4d34-b784-9b728131a338\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>0.3</td>\n",
              "      <td>286.6</td>\n",
              "      <td>14-3-3 protein-like protein [Leishmania paname...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.000000e-86</td>\n",
              "      <td>AIO02465.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>1.1</td>\n",
              "      <td>283.2</td>\n",
              "      <td>14-3-3 protein, putative [Leishmania panamensis]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.400000e-85</td>\n",
              "      <td>AIN96231.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.2</td>\n",
              "      <td>12.3</td>\n",
              "      <td>hypothetical protein LPMP_180840 [Leishmania p...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.8</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3.900000e-02</td>\n",
              "      <td>AIN97265.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.1</td>\n",
              "      <td>121.6</td>\n",
              "      <td>D-3-phosphoglycerate dehydrogenase-like protei...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>7.200000e-36</td>\n",
              "      <td>AIN95237.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.1</td>\n",
              "      <td>181.5</td>\n",
              "      <td>D-3-phosphoglycerate dehydrogenase-like protei...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.3</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.900000e-54</td>\n",
              "      <td>AIN95237.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3632</th>\n",
              "      <td>-</td>\n",
              "      <td>8.1</td>\n",
              "      <td>21.4</td>\n",
              "      <td>hypothetical protein LPMP_321160 [Leishmania p...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>7.000000e-05</td>\n",
              "      <td>AIO01002.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3633</th>\n",
              "      <td>-</td>\n",
              "      <td>9.7</td>\n",
              "      <td>18.8</td>\n",
              "      <td>hypothetical protein LPMP_324060 [Leishmania p...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.7</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.600000e-04</td>\n",
              "      <td>AIO01287.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3634</th>\n",
              "      <td>-</td>\n",
              "      <td>3.4</td>\n",
              "      <td>12.6</td>\n",
              "      <td>dual specificity protein phosphatase, putative...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.000000e-02</td>\n",
              "      <td>AIN97681.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3635</th>\n",
              "      <td>-</td>\n",
              "      <td>22.8</td>\n",
              "      <td>11.2</td>\n",
              "      <td>protein kinase, putative [Leishmania panamensis]</td>\n",
              "      <td>0</td>\n",
              "      <td>2.7</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>1.000000e-01</td>\n",
              "      <td>AIN98802.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3636</th>\n",
              "      <td>-</td>\n",
              "      <td>9.0</td>\n",
              "      <td>11.2</td>\n",
              "      <td>ubiquitin-protein ligase-like, putative [Leish...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.5</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.100000e-01</td>\n",
              "      <td>AIN95717.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3637 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-eb7892d4-b8f6-4d34-b784-9b728131a338')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-eb7892d4-b8f6-4d34-b784-9b728131a338 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-eb7892d4-b8f6-4d34-b784-9b728131a338');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-9a107949-65a1-41eb-a2a9-7515101232f5\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-9a107949-65a1-41eb-a2a9-7515101232f5')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-9a107949-65a1-41eb-a2a9-7515101232f5 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 3637,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.512553886517209,\n        \"min\": 0.0,\n        \"max\": 196.2,\n        \"num_unique_values\": 316,\n        \"samples\": [\n          26.9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 66.60081016850296,\n        \"min\": 3.6,\n        \"max\": 870.8,\n        \"num_unique_values\": 1246,\n        \"samples\": [\n          50.6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1516,\n        \"samples\": [\n          \"hypothetical protein LPMP_331260 [Leishmania panamensis]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 6,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.328626166607032,\n        \"min\": 1.0,\n        \"max\": 24.2,\n        \"num_unique_values\": 139,\n        \"samples\": [\n          15.9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 15,\n        \"num_unique_values\": 16,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 24,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.6151191555526765,\n        \"min\": 1.7e-262,\n        \"max\": 9.6,\n        \"num_unique_values\": 1828,\n        \"samples\": [\n          3.7e-08\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1911,\n        \"samples\": [\n          \"AIN98043.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 312,\n        \"samples\": [\n          \"SIR2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 8,\n        \"num_unique_values\": 9,\n        \"samples\": [\n          6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 23,\n        \"samples\": [\n          14\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 161
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"leishmania_panamensi_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "Yd4iBSEep2xr"
      },
      "execution_count": 162,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_000755165.1_ASM75516v1_genomic.fna -bed leishmania_panamensi_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "id": "eoVsZqQIp5Qx"
      },
      "execution_count": 163,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_leishmania_panamensi_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "xhNEjvhlp7QZ"
      },
      "execution_count": 164,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import random\n",
        "f = open(\"Angomonas_filtered_genes.bed\", \"w\")\n",
        "count_hmmer_genes = 0\n",
        "\n",
        "while count_hmmer_genes != 4:\n",
        "  index = random.randint(0, len(hmmer_data))\n",
        "  protein = hmmer_data.iloc[[index]]\n",
        "  gene = genome.loc[genome[\"attributes\"].str.contains(protein[\"id\"].item())]\n",
        "  while gene[\"type\"].item() != \"gene\":\n",
        "    gene = genome.iloc[[gene.index.item() - 1]]\n",
        "  if not ((gene[\"end\"].item() - gene[\"start\"].item()) <= 2000):\n",
        "    continue\n",
        "  f.write(f'{gene.iloc[[0]][\"chrom\"].item()}\\t{gene.iloc[[0]][\"start\"].item()}\\t{gene.iloc[[0]][\"end\"].item()}\\t{protein[\"id\"].item()}\\n')\n",
        "  count_hmmer_genes += 1\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "Q-vbUvwjU6Ll"
      },
      "execution_count": 169,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_000755165.1_ASM75516v1_genomic.fna -bed Angomonas_filtered_genes.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "id": "xaNJQBv0U6Lm"
      },
      "execution_count": 170,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Leishmania donovani"
      ],
      "metadata": {
        "id": "vcjwe1vjrQo0"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "cK6DGlC9rp2c"
      },
      "execution_count": 80,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Leishmania_donovani_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 930
        },
        "outputId": "fc82309b-5fe2-4c33-e4d6-93781fbfcdbd",
        "id": "wAu5tB3Arp2c"
      },
      "execution_count": 81,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   1.2     288.5   \n",
              "1            -   1.8     284.3   \n",
              "2            -   0.1      11.0   \n",
              "3            -   0.1     120.8   \n",
              "4            -   0.0      15.3   \n",
              "...        ...   ...       ...   \n",
              "3690         -   8.7      18.9   \n",
              "3691         -   8.4      12.0   \n",
              "3692         -   8.6       9.3   \n",
              "3693         -  22.1       8.8   \n",
              "3694         -  12.1       5.1   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0     14-3-3 protein-like protein [Leishmania donovani]            0   \n",
              "1        14-3-3 protein, putative [Leishmania donovani]            0   \n",
              "2     hypothetical protein, conserved [Leishmania do...            0   \n",
              "3     D-3-phosphoglycerate dehydrogenase-like protei...            0   \n",
              "4     unnamed protein product, partial [Leishmania d...            1   \n",
              "...                                                 ...          ...   \n",
              "3690  hypothetical protein, conserved [Leishmania do...            0   \n",
              "3691  ubiquitin-protein ligase-like, putative [Leish...            0   \n",
              "3692  hypothetical protein, conserved [Leishmania do...            0   \n",
              "3693     protein kinase, putative [Leishmania donovani]            0   \n",
              "3694  hypothetical protein, conserved [Leishmania do...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                2.9                    0               3   \n",
              "3                1.1                    1               1   \n",
              "4                1.5                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "3690             1.6                    1               1   \n",
              "3691             1.5                    0               1   \n",
              "3692             1.8                    0               1   \n",
              "3693             2.6                    0               2   \n",
              "3694             3.0                    0               3   \n",
              "\n",
              "      domain_reported_num  env_num        evalue          id    query_id  \\\n",
              "0                       1        1  1.400000e-86  CBZ38774.1      14-3-3   \n",
              "1                       1        1  2.600000e-85  CBZ32298.1      14-3-3   \n",
              "2                       3        3  1.000000e-01  CBZ33390.1      14-3-3   \n",
              "3                       1        1  1.300000e-35  CBZ31244.1  2-Hacid_dh   \n",
              "4                       1        1  5.100000e-03  CBZ37913.1  2-Hacid_dh   \n",
              "...                   ...      ...           ...         ...         ...   \n",
              "3690                    1        1  4.300000e-04  CBZ37135.1          ZZ   \n",
              "3691                    1        1  6.300000e-02  CBZ31762.1          ZZ   \n",
              "3692                    1        1  4.400000e-01  CBZ38149.1          ZZ   \n",
              "3693                    2        2  6.400000e-01  CBZ34568.1          ZZ   \n",
              "3694                    3        3  9.000000e+00  CBZ37832.1          ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           3  \n",
              "3               0           1  \n",
              "4               0           1  \n",
              "...           ...         ...  \n",
              "3690            0           1  \n",
              "3691            0           1  \n",
              "3692            0           1  \n",
              "3693            0           2  \n",
              "3694            0           3  \n",
              "\n",
              "[3695 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1804ab2c-7114-4094-882d-adedbcd2f051\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>1.2</td>\n",
              "      <td>288.5</td>\n",
              "      <td>14-3-3 protein-like protein [Leishmania donovani]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.400000e-86</td>\n",
              "      <td>CBZ38774.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>1.8</td>\n",
              "      <td>284.3</td>\n",
              "      <td>14-3-3 protein, putative [Leishmania donovani]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.600000e-85</td>\n",
              "      <td>CBZ32298.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.1</td>\n",
              "      <td>11.0</td>\n",
              "      <td>hypothetical protein, conserved [Leishmania do...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.9</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1.000000e-01</td>\n",
              "      <td>CBZ33390.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.1</td>\n",
              "      <td>120.8</td>\n",
              "      <td>D-3-phosphoglycerate dehydrogenase-like protei...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.300000e-35</td>\n",
              "      <td>CBZ31244.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>15.3</td>\n",
              "      <td>unnamed protein product, partial [Leishmania d...</td>\n",
              "      <td>1</td>\n",
              "      <td>1.5</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.100000e-03</td>\n",
              "      <td>CBZ37913.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3690</th>\n",
              "      <td>-</td>\n",
              "      <td>8.7</td>\n",
              "      <td>18.9</td>\n",
              "      <td>hypothetical protein, conserved [Leishmania do...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.300000e-04</td>\n",
              "      <td>CBZ37135.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3691</th>\n",
              "      <td>-</td>\n",
              "      <td>8.4</td>\n",
              "      <td>12.0</td>\n",
              "      <td>ubiquitin-protein ligase-like, putative [Leish...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.5</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>6.300000e-02</td>\n",
              "      <td>CBZ31762.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3692</th>\n",
              "      <td>-</td>\n",
              "      <td>8.6</td>\n",
              "      <td>9.3</td>\n",
              "      <td>hypothetical protein, conserved [Leishmania do...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.400000e-01</td>\n",
              "      <td>CBZ38149.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3693</th>\n",
              "      <td>-</td>\n",
              "      <td>22.1</td>\n",
              "      <td>8.8</td>\n",
              "      <td>protein kinase, putative [Leishmania donovani]</td>\n",
              "      <td>0</td>\n",
              "      <td>2.6</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>6.400000e-01</td>\n",
              "      <td>CBZ34568.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3694</th>\n",
              "      <td>-</td>\n",
              "      <td>12.1</td>\n",
              "      <td>5.1</td>\n",
              "      <td>hypothetical protein, conserved [Leishmania do...</td>\n",
              "      <td>0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>9.000000e+00</td>\n",
              "      <td>CBZ37832.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3695 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1804ab2c-7114-4094-882d-adedbcd2f051')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1804ab2c-7114-4094-882d-adedbcd2f051 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1804ab2c-7114-4094-882d-adedbcd2f051');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-03ab3c8f-4c52-44a2-aa55-1b8c28178f10\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-03ab3c8f-4c52-44a2-aa55-1b8c28178f10')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-03ab3c8f-4c52-44a2-aa55-1b8c28178f10 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 3695,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.266702187662919,\n        \"min\": 0.0,\n        \"max\": 201.3,\n        \"num_unique_values\": 309,\n        \"samples\": [\n          16.4\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 65.01140999680551,\n        \"min\": 3.5,\n        \"max\": 872.5,\n        \"num_unique_values\": 1238,\n        \"samples\": [\n          5.3\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 589,\n        \"samples\": [\n          \"protein transport protein Sec31, putative, partial [Leishmania donovani]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 6,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.3045567928717365,\n        \"min\": 1.0,\n        \"max\": 24.4,\n        \"num_unique_values\": 135,\n        \"samples\": [\n          9.2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 16,\n        \"num_unique_values\": 16,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          19\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 25,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          19\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.7922629625725888,\n        \"min\": 5.5e-263,\n        \"max\": 9.1,\n        \"num_unique_values\": 1872,\n        \"samples\": [\n          2.5e-47\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1946,\n        \"samples\": [\n          \"CBZ35966.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 323,\n        \"samples\": [\n          \"OCRE\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 10,\n        \"num_unique_values\": 9,\n        \"samples\": [\n          5\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          18\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 81
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Leishmania_donovani_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "3YjtZ_Xurp2d"
      },
      "execution_count": 82,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_000227135.2_ASM22713v2_genomic.fna -bed Leishmania_donovani_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d6b8dead-0043-4908-f544-cb087f3f76aa",
        "id": "pYtueN5Brp2d"
      },
      "execution_count": 84,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file GCA_000227135.2_ASM22713v2_genomic.fna.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_leishmania_donovani_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "Ub2DoqNRrp2d"
      },
      "execution_count": 85,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Leishmania martiniquensis"
      ],
      "metadata": {
        "id": "pZLnMgRI2x-x"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "kSvpcF893EKZ"
      },
      "execution_count": 121,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Leishmania_martiniquensis_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "5cde0765-a609-4f21-b206-ade8f36e6ac0",
        "id": "N2AmYZrV3EKZ"
      },
      "execution_count": 123,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   1.1     290.1   \n",
              "1            -   4.1     283.7   \n",
              "2            -   0.0      16.7   \n",
              "3            -   0.0     151.2   \n",
              "4            -   0.0      37.9   \n",
              "...        ...   ...       ...   \n",
              "3714         -  17.8      18.2   \n",
              "3715         -   9.0      18.0   \n",
              "3716         -  11.1      15.6   \n",
              "3717         -   7.0      12.2   \n",
              "3718         -  27.2       7.2   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0     hypothetical protein LSCM1_00343 [Leishmania m...            0   \n",
              "1     hypothetical protein LSCM1_06546 [Leishmania m...            0   \n",
              "2     hypothetical protein LSCM1_02201 [Leishmania m...            0   \n",
              "3     hypothetical protein LSCM1_02201 [Leishmania m...            0   \n",
              "4     hypothetical protein LSCM1_00420 [Leishmania m...            0   \n",
              "...                                                 ...          ...   \n",
              "3714  hypothetical protein LSCM1_05133 [Leishmania m...            0   \n",
              "3715  hypothetical protein LSCM1_01753 [Leishmania m...            0   \n",
              "3716  hypothetical protein LSCM1_02051 [Leishmania m...            0   \n",
              "3717  hypothetical protein LSCM1_07284 [Leishmania m...            0   \n",
              "3718  hypothetical protein LSCM1_06371 [Leishmania m...            1   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                1.2                    1               1   \n",
              "3                1.2                    1               1   \n",
              "4                1.4                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "3714             2.5                    2               2   \n",
              "3715             2.0                    1               2   \n",
              "3716             1.6                    1               1   \n",
              "3717             1.5                    0               1   \n",
              "3718             2.8                    0               2   \n",
              "\n",
              "      domain_reported_num  env_num        evalue            id      query_id  \\\n",
              "0                       1        1  4.500000e-87  KAG5464163.1        14-3-3   \n",
              "1                       1        1  3.900000e-85  KAG5484725.1        14-3-3   \n",
              "2                       1        1  1.900000e-03  KAG5468224.1    2-Hacid_dh   \n",
              "3                       1        1  8.000000e-45  KAG5468224.1  2-Hacid_dh_C   \n",
              "4                       1        1  4.900000e-10  KAG5464239.1  2-Hacid_dh_C   \n",
              "...                   ...      ...           ...           ...           ...   \n",
              "3714                    2        2  6.900000e-04  KAG5477831.1            ZZ   \n",
              "3715                    2        2  8.100000e-04  KAG5470509.1            ZZ   \n",
              "3716                    1        1  4.700000e-03  KAG5470801.1            ZZ   \n",
              "3717                    1        1  5.200000e-02  KAG5486166.1            ZZ   \n",
              "3718                    2        2  1.900000e+00  KAG5479952.1            ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           1  \n",
              "3               0           1  \n",
              "4               0           1  \n",
              "...           ...         ...  \n",
              "3714            0           2  \n",
              "3715            0           2  \n",
              "3716            0           1  \n",
              "3717            0           1  \n",
              "3718            0           2  \n",
              "\n",
              "[3719 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-b118f7d5-b547-4834-8eea-b469078dc582\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>1.1</td>\n",
              "      <td>290.1</td>\n",
              "      <td>hypothetical protein LSCM1_00343 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.500000e-87</td>\n",
              "      <td>KAG5464163.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>4.1</td>\n",
              "      <td>283.7</td>\n",
              "      <td>hypothetical protein LSCM1_06546 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.900000e-85</td>\n",
              "      <td>KAG5484725.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>16.7</td>\n",
              "      <td>hypothetical protein LSCM1_02201 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.900000e-03</td>\n",
              "      <td>KAG5468224.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>151.2</td>\n",
              "      <td>hypothetical protein LSCM1_02201 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>8.000000e-45</td>\n",
              "      <td>KAG5468224.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>37.9</td>\n",
              "      <td>hypothetical protein LSCM1_00420 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.900000e-10</td>\n",
              "      <td>KAG5464239.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3714</th>\n",
              "      <td>-</td>\n",
              "      <td>17.8</td>\n",
              "      <td>18.2</td>\n",
              "      <td>hypothetical protein LSCM1_05133 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.5</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>6.900000e-04</td>\n",
              "      <td>KAG5477831.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3715</th>\n",
              "      <td>-</td>\n",
              "      <td>9.0</td>\n",
              "      <td>18.0</td>\n",
              "      <td>hypothetical protein LSCM1_01753 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.0</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>8.100000e-04</td>\n",
              "      <td>KAG5470509.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3716</th>\n",
              "      <td>-</td>\n",
              "      <td>11.1</td>\n",
              "      <td>15.6</td>\n",
              "      <td>hypothetical protein LSCM1_02051 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.700000e-03</td>\n",
              "      <td>KAG5470801.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3717</th>\n",
              "      <td>-</td>\n",
              "      <td>7.0</td>\n",
              "      <td>12.2</td>\n",
              "      <td>hypothetical protein LSCM1_07284 [Leishmania m...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.5</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.200000e-02</td>\n",
              "      <td>KAG5486166.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3718</th>\n",
              "      <td>-</td>\n",
              "      <td>27.2</td>\n",
              "      <td>7.2</td>\n",
              "      <td>hypothetical protein LSCM1_06371 [Leishmania m...</td>\n",
              "      <td>1</td>\n",
              "      <td>2.8</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>1.900000e+00</td>\n",
              "      <td>KAG5479952.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3719 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-b118f7d5-b547-4834-8eea-b469078dc582')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-b118f7d5-b547-4834-8eea-b469078dc582 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-b118f7d5-b547-4834-8eea-b469078dc582');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-f443b7d8-93b4-4bb7-b0ef-69ab560e9603\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-f443b7d8-93b4-4bb7-b0ef-69ab560e9603')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-f443b7d8-93b4-4bb7-b0ef-69ab560e9603 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 3719,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.34916661806384,\n        \"min\": 0.0,\n        \"max\": 208.3,\n        \"num_unique_values\": 308,\n        \"samples\": [\n          10.9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 58.91417028143694,\n        \"min\": 3.2,\n        \"max\": 685.5,\n        \"num_unique_values\": 1227,\n        \"samples\": [\n          56.5\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1923,\n        \"samples\": [\n          \"hypothetical protein LSCM1_06310 [Leishmania martiniquensis]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 5,\n        \"num_unique_values\": 6,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.3828653770743844,\n        \"min\": 1.0,\n        \"max\": 24.6,\n        \"num_unique_values\": 140,\n        \"samples\": [\n          11.3\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 15,\n        \"num_unique_values\": 16,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 28,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          19\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 25,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 28,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          19\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.7306476944976481,\n        \"min\": 3.1e-206,\n        \"max\": 9.9,\n        \"num_unique_values\": 1831,\n        \"samples\": [\n          0.087\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1923,\n        \"samples\": [\n          \"KAG5480606.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 312,\n        \"samples\": [\n          \"SNF5\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 10,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          5\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 27,\n        \"num_unique_values\": 24,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 123
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Leishmania_martiniquensis_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "g6G9-_Ge3EKZ"
      },
      "execution_count": 124,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_017916325.1_LU_Lmar_1.0_genomic.fna -bed Leishmania_martiniquensis_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "99679042-bdef-46e7-8029-1cc11c512393",
        "id": "mXFBck083EKZ"
      },
      "execution_count": 125,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file GCA_017916325.1_LU_Lmar_1.0_genomic.fna.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_Leishmania_martiniquensis_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "BxRU88jA3EKZ"
      },
      "execution_count": 126,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Leptomonas pyrrhocoris"
      ],
      "metadata": {
        "id": "t0LLtseTtr6r"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "TFWKs9qYtzOx"
      },
      "execution_count": 115,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Leptomonas_pyrrhocoris_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 860
        },
        "outputId": "3813ec41-ef9a-41ff-e51d-41b3c4baafe6",
        "id": "tn3q-5ZhtzOx"
      },
      "execution_count": 116,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   1.7     288.2   \n",
              "1            -   1.7     288.2   \n",
              "2            -   3.6     286.4   \n",
              "3            -   0.2      16.3   \n",
              "4            -   4.5      11.8   \n",
              "...        ...   ...       ...   \n",
              "5925         -  21.4      10.2   \n",
              "5926         -   3.8      10.0   \n",
              "5927         -   8.0       7.2   \n",
              "5928         -  18.8       6.7   \n",
              "5929         -  18.8       6.7   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0     14-3-3 protein-like protein [Leptomonas pyrrho...            0   \n",
              "1     14-3-3 protein-like protein [Leptomonas pyrrho...            0   \n",
              "2      putative 14-3-3 protein [Leptomonas pyrrhocoris]            0   \n",
              "3     hypothetical protein ABB37_06165 [Leptomonas p...            0   \n",
              "4     hypothetical protein ABB37_09465 [Leptomonas p...            1   \n",
              "...                                                 ...          ...   \n",
              "5925  putative calcium/calmodulin-dependent protein ...            0   \n",
              "5926  putative dual specificity protein phosphatase ...            0   \n",
              "5927  hypothetical protein ABB37_00458 [Leptomonas p...            0   \n",
              "5928  hypothetical protein ABB37_03568 [Leptomonas p...            0   \n",
              "5929  hypothetical protein ABB37_03568 [Leptomonas p...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                1.0                    1               1   \n",
              "3                3.1                    2               3   \n",
              "4                2.6                    0               3   \n",
              "...              ...                  ...             ...   \n",
              "5925             2.5                    0               2   \n",
              "5926             1.4                    0               1   \n",
              "5927             1.5                    0               1   \n",
              "5928             1.8                    0               2   \n",
              "5929             1.8                    0               2   \n",
              "\n",
              "      domain_reported_num  env_num        evalue              id query_id  \\\n",
              "0                       1        1  2.900000e-86  XP_015659743.1   14-3-3   \n",
              "1                       1        1  2.900000e-86  XP_015659744.1   14-3-3   \n",
              "2                       1        1  1.000000e-85  XP_015655378.1   14-3-3   \n",
              "3                       3        3  4.200000e-03  XP_015657004.1   14-3-3   \n",
              "4                       3        3  1.000000e-01  XP_015652651.1   14-3-3   \n",
              "...                   ...      ...           ...             ...      ...   \n",
              "5925                    2        2  3.800000e-01  XP_015657701.1       ZZ   \n",
              "5926                    1        1  4.400000e-01  XP_015660705.1       ZZ   \n",
              "5927                    1        1  3.400000e+00  XP_015664660.1       ZZ   \n",
              "5928                    1        2  4.900000e+00  XP_015660958.1       ZZ   \n",
              "5929                    1        2  4.900000e+00  XP_015660959.1       ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           1  \n",
              "3               0           3  \n",
              "4               2           1  \n",
              "...           ...         ...  \n",
              "5925            0           2  \n",
              "5926            0           1  \n",
              "5927            0           1  \n",
              "5928            0           2  \n",
              "5929            0           2  \n",
              "\n",
              "[5930 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8fde2f85-5780-4712-ab05-767d794150c6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>1.7</td>\n",
              "      <td>288.2</td>\n",
              "      <td>14-3-3 protein-like protein [Leptomonas pyrrho...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.900000e-86</td>\n",
              "      <td>XP_015659743.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>1.7</td>\n",
              "      <td>288.2</td>\n",
              "      <td>14-3-3 protein-like protein [Leptomonas pyrrho...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.900000e-86</td>\n",
              "      <td>XP_015659744.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>3.6</td>\n",
              "      <td>286.4</td>\n",
              "      <td>putative 14-3-3 protein [Leptomonas pyrrhocoris]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.000000e-85</td>\n",
              "      <td>XP_015655378.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.2</td>\n",
              "      <td>16.3</td>\n",
              "      <td>hypothetical protein ABB37_06165 [Leptomonas p...</td>\n",
              "      <td>0</td>\n",
              "      <td>3.1</td>\n",
              "      <td>2</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>4.200000e-03</td>\n",
              "      <td>XP_015657004.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>4.5</td>\n",
              "      <td>11.8</td>\n",
              "      <td>hypothetical protein ABB37_09465 [Leptomonas p...</td>\n",
              "      <td>1</td>\n",
              "      <td>2.6</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1.000000e-01</td>\n",
              "      <td>XP_015652651.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5925</th>\n",
              "      <td>-</td>\n",
              "      <td>21.4</td>\n",
              "      <td>10.2</td>\n",
              "      <td>putative calcium/calmodulin-dependent protein ...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.5</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>3.800000e-01</td>\n",
              "      <td>XP_015657701.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5926</th>\n",
              "      <td>-</td>\n",
              "      <td>3.8</td>\n",
              "      <td>10.0</td>\n",
              "      <td>putative dual specificity protein phosphatase ...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.400000e-01</td>\n",
              "      <td>XP_015660705.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5927</th>\n",
              "      <td>-</td>\n",
              "      <td>8.0</td>\n",
              "      <td>7.2</td>\n",
              "      <td>hypothetical protein ABB37_00458 [Leptomonas p...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.5</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.400000e+00</td>\n",
              "      <td>XP_015664660.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5928</th>\n",
              "      <td>-</td>\n",
              "      <td>18.8</td>\n",
              "      <td>6.7</td>\n",
              "      <td>hypothetical protein ABB37_03568 [Leptomonas p...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>4.900000e+00</td>\n",
              "      <td>XP_015660958.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5929</th>\n",
              "      <td>-</td>\n",
              "      <td>18.8</td>\n",
              "      <td>6.7</td>\n",
              "      <td>hypothetical protein ABB37_03568 [Leptomonas p...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>4.900000e+00</td>\n",
              "      <td>XP_015660959.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5930 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8fde2f85-5780-4712-ab05-767d794150c6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-8fde2f85-5780-4712-ab05-767d794150c6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-8fde2f85-5780-4712-ab05-767d794150c6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-3eab8b6f-cb97-4193-9b27-c357198a1980\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-3eab8b6f-cb97-4193-9b27-c357198a1980')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-3eab8b6f-cb97-4193-9b27-c357198a1980 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 5930,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.862311460530428,\n        \"min\": 0.0,\n        \"max\": 194.8,\n        \"num_unique_values\": 342,\n        \"samples\": [\n          20.1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 81.90317490730628,\n        \"min\": 4.0,\n        \"max\": 873.2,\n        \"num_unique_values\": 1299,\n        \"samples\": [\n          85.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1743,\n        \"samples\": [\n          \"putative serine/threonine-protein kinase a [Leptomonas pyrrhocoris]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 6,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.5158965188258544,\n        \"min\": 1.0,\n        \"max\": 26.6,\n        \"num_unique_values\": 149,\n        \"samples\": [\n          10.1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 19,\n        \"num_unique_values\": 17,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 29,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          20\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 28,\n        \"num_unique_values\": 20,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 29,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          20\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.759289366458412,\n        \"min\": 5.6e-263,\n        \"max\": 10.0,\n        \"num_unique_values\": 1972,\n        \"samples\": [\n          1.5e-15\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 3116,\n        \"samples\": [\n          \"XP_015656146.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 321,\n        \"samples\": [\n          \"OB_NTP_bind\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 15,\n        \"num_unique_values\": 11,\n        \"samples\": [\n          4\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 29,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          20\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 116
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Leptomonas_pyrrhocoris_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5WlKcYl3tzOx",
        "outputId": "eef93d1c-5766-487e-bf6f-b55dd5b9cf60"
      },
      "execution_count": 117,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "AAA\n",
            "Helicase_C\n",
            "Metallophos\n",
            "Methyltransf_11\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCF_001293395.1_ASM129339v1_genomic.fna -bed Leptomonas_pyrrhocoris_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "21cf7b59-f26b-43c3-f0b2-54b73b5e77fe",
        "id": "0m6nlvq3tzOy"
      },
      "execution_count": 119,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file GCF_001293395.1_ASM129339v1_genomic.fna.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_Leptomonas_pyrrhocoris_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "E4tp-9wLtzOy"
      },
      "execution_count": 120,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Novymonas esmeraldas"
      ],
      "metadata": {
        "id": "6SEuI_hl5Nld"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "bFgG6jBo5YMP"
      },
      "execution_count": 127,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Novymonas esmeraldas_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 843
        },
        "outputId": "be136a86-7442-4e48-9c5a-7a9b19b73748",
        "id": "RwuNU19R5YMQ"
      },
      "execution_count": 129,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   1.4     289.5   \n",
              "1            -   2.4     284.9   \n",
              "2            -   0.0      24.2   \n",
              "3            -   0.0      20.7   \n",
              "4            -   0.0      18.8   \n",
              "...        ...   ...       ...   \n",
              "3996         -   7.7      13.8   \n",
              "3997         -  10.5      11.0   \n",
              "3998         -  22.6       8.2   \n",
              "3999         -   5.5       7.3   \n",
              "4000         -  11.3       6.6   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0               14-3-3 protein 1 [Novymonas esmeraldas]            0   \n",
              "1               14-3-3 protein 2 [Novymonas esmeraldas]            0   \n",
              "2     d-isomer specific 2-hydroxyacid dehydrogenase-...            0   \n",
              "3     d-isomer specific 2-hydroxyacid dehydrogenase-...            1   \n",
              "4     d-isomer specific 2-hydroxyacid dehydrogenase-...            0   \n",
              "...                                                 ...          ...   \n",
              "3996  ubiquitin-protein ligase-like [Novymonas esmer...            0   \n",
              "3997  SPRY domain/HECT-domain (ubiquitin-transferase...            0   \n",
              "3998  calcium/calmodulin-dependent protein kinase [N...            0   \n",
              "3999  dual specificity protein phosphatase [Novymona...            0   \n",
              "4000  hypothetical protein NESM_000523000 [Novymonas...            1   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                1.2                    1               1   \n",
              "3                1.4                    1               1   \n",
              "4                1.2                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "3996             1.6                    0               1   \n",
              "3997             1.6                    0               1   \n",
              "3998             2.6                    0               2   \n",
              "3999             1.4                    0               1   \n",
              "4000             3.0                    0               3   \n",
              "\n",
              "      domain_reported_num  env_num        evalue            id    query_id  \\\n",
              "0                       1        1  7.100000e-87  KAK7197891.1      14-3-3   \n",
              "1                       1        1  1.800000e-85  KAK7200685.1      14-3-3   \n",
              "2                       1        1  9.700000e-06  KAK7199519.1  2-Hacid_dh   \n",
              "3                       1        1  1.200000e-04  KAK7200793.1  2-Hacid_dh   \n",
              "4                       1        1  4.400000e-04  KAK7200794.1  2-Hacid_dh   \n",
              "...                   ...      ...           ...           ...         ...   \n",
              "3996                    1        1  1.800000e-02  KAK7194481.1          ZZ   \n",
              "3997                    1        1  1.300000e-01  KAK7196112.1          ZZ   \n",
              "3998                    2        2  9.800000e-01  KAK7199000.1          ZZ   \n",
              "3999                    1        1  1.900000e+00  KAK7200755.1          ZZ   \n",
              "4000                    3        3  3.100000e+00  KAK7195903.1          ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           1  \n",
              "3               0           1  \n",
              "4               0           1  \n",
              "...           ...         ...  \n",
              "3996            0           1  \n",
              "3997            0           1  \n",
              "3998            0           2  \n",
              "3999            0           1  \n",
              "4000            1           2  \n",
              "\n",
              "[4001 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-c9ca3f58-1d07-4027-ae8e-c38bebdb3e6a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>1.4</td>\n",
              "      <td>289.5</td>\n",
              "      <td>14-3-3 protein 1 [Novymonas esmeraldas]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>7.100000e-87</td>\n",
              "      <td>KAK7197891.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>2.4</td>\n",
              "      <td>284.9</td>\n",
              "      <td>14-3-3 protein 2 [Novymonas esmeraldas]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.800000e-85</td>\n",
              "      <td>KAK7200685.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>24.2</td>\n",
              "      <td>d-isomer specific 2-hydroxyacid dehydrogenase-...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>9.700000e-06</td>\n",
              "      <td>KAK7199519.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>20.7</td>\n",
              "      <td>d-isomer specific 2-hydroxyacid dehydrogenase-...</td>\n",
              "      <td>1</td>\n",
              "      <td>1.4</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.200000e-04</td>\n",
              "      <td>KAK7200793.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>18.8</td>\n",
              "      <td>d-isomer specific 2-hydroxyacid dehydrogenase-...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.2</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.400000e-04</td>\n",
              "      <td>KAK7200794.1</td>\n",
              "      <td>2-Hacid_dh</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3996</th>\n",
              "      <td>-</td>\n",
              "      <td>7.7</td>\n",
              "      <td>13.8</td>\n",
              "      <td>ubiquitin-protein ligase-like [Novymonas esmer...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.800000e-02</td>\n",
              "      <td>KAK7194481.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3997</th>\n",
              "      <td>-</td>\n",
              "      <td>10.5</td>\n",
              "      <td>11.0</td>\n",
              "      <td>SPRY domain/HECT-domain (ubiquitin-transferase...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.300000e-01</td>\n",
              "      <td>KAK7196112.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3998</th>\n",
              "      <td>-</td>\n",
              "      <td>22.6</td>\n",
              "      <td>8.2</td>\n",
              "      <td>calcium/calmodulin-dependent protein kinase [N...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.6</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>9.800000e-01</td>\n",
              "      <td>KAK7199000.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3999</th>\n",
              "      <td>-</td>\n",
              "      <td>5.5</td>\n",
              "      <td>7.3</td>\n",
              "      <td>dual specificity protein phosphatase [Novymona...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.900000e+00</td>\n",
              "      <td>KAK7200755.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4000</th>\n",
              "      <td>-</td>\n",
              "      <td>11.3</td>\n",
              "      <td>6.6</td>\n",
              "      <td>hypothetical protein NESM_000523000 [Novymonas...</td>\n",
              "      <td>1</td>\n",
              "      <td>3.0</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3.100000e+00</td>\n",
              "      <td>KAK7195903.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>4001 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-c9ca3f58-1d07-4027-ae8e-c38bebdb3e6a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-c9ca3f58-1d07-4027-ae8e-c38bebdb3e6a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-c9ca3f58-1d07-4027-ae8e-c38bebdb3e6a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-9d2ec3bb-0d05-4691-879f-d594f620b45f\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-9d2ec3bb-0d05-4691-879f-d594f620b45f')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-9d2ec3bb-0d05-4691-879f-d594f620b45f button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 4001,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.363289115502356,\n        \"min\": 0.0,\n        \"max\": 226.9,\n        \"num_unique_values\": 316,\n        \"samples\": [\n          22.6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 61.444863789190975,\n        \"min\": 3.1,\n        \"max\": 876.9,\n        \"num_unique_values\": 1272,\n        \"samples\": [\n          9.9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1359,\n        \"samples\": [\n          \"replication factor C, subunit 3 [Novymonas esmeraldas]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 6,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.7313809780014675,\n        \"min\": 1.0,\n        \"max\": 27.0,\n        \"num_unique_values\": 156,\n        \"samples\": [\n          24.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 15,\n        \"num_unique_values\": 16,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 30,\n        \"num_unique_values\": 27,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 28,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          10\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 30,\n        \"num_unique_values\": 27,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.6947088118365867,\n        \"min\": 2.5e-264,\n        \"max\": 9.6,\n        \"num_unique_values\": 1902,\n        \"samples\": [\n          8.7e-58\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2046,\n        \"samples\": [\n          \"KAK7194942.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 312,\n        \"samples\": [\n          \"SHNi-TPR\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 15,\n        \"num_unique_values\": 13,\n        \"samples\": [\n          9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 28,\n        \"num_unique_values\": 25,\n        \"samples\": [\n          8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 129
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Novymonas_esmeraldas_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "W6esaKmk5YMQ"
      },
      "execution_count": 130,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_019188245.1_ASM1918824v1_genomic.fna -bed Novymonas_esmeraldas_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9ddedffb-b8b0-4f47-b5cc-a276f3f32e1c",
        "id": "vRLDK2Za5YMQ"
      },
      "execution_count": 131,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file GCA_019188245.1_ASM1918824v1_genomic.fna.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_Novymonas_esmeraldas_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "oQJJvBsm5YMQ"
      },
      "execution_count": 132,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Trypanosoma rangeli"
      ],
      "metadata": {
        "id": "tKV5hgQI7ieE"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "0beAOFHr73d0"
      },
      "execution_count": 139,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Trypanosoma_rangeli_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 860
        },
        "outputId": "a05e8d06-bef1-4a30-a9c0-c82b5a3dc8f7",
        "id": "9vu1e8ts73d0"
      },
      "execution_count": 140,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   5.6     298.5   \n",
              "1            -   1.9     293.6   \n",
              "2            -   0.0      17.9   \n",
              "3            -   0.0     153.4   \n",
              "4            -   0.0     149.3   \n",
              "...        ...   ...       ...   \n",
              "3860         -   8.1      17.8   \n",
              "3861         -  11.8      17.1   \n",
              "3862         -   1.5      12.5   \n",
              "3863         -  11.1       6.3   \n",
              "3864         -  13.1       5.5   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0                  14-3-3 protein [Trypanosoma rangeli]            0   \n",
              "1                  14-3-3 protein [Trypanosoma rangeli]            0   \n",
              "2     uncharacterized protein TraAM80_00771 [Trypano...            0   \n",
              "3     D-isomer specific 2-hydroxyacid dehydrogenase-...            0   \n",
              "4     D-isomer specific 2-hydroxyacid dehydrogenase-...            0   \n",
              "...                                                 ...          ...   \n",
              "3860  E3 ubiquitin-protein ligase TRIP12 [Trypanosom...            0   \n",
              "3861  uncharacterized protein TraAM80_04771 [Trypano...            0   \n",
              "3862  ADP-ribosylation factor GTPase activating prot...            0   \n",
              "3863      putative protein kinase [Trypanosoma rangeli]            0   \n",
              "3864  other hect domain ubiquitin protein ligase E3 ...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.1                    1               1   \n",
              "2                2.5                    2               2   \n",
              "3                1.0                    1               1   \n",
              "4                1.1                    1               1   \n",
              "...              ...                  ...             ...   \n",
              "3860             1.6                    1               1   \n",
              "3861             1.8                    1               1   \n",
              "3862             1.6                    0               1   \n",
              "3863             2.0                    0               2   \n",
              "3864             1.8                    0               2   \n",
              "\n",
              "      domain_reported_num  env_num        evalue              id  \\\n",
              "0                       1        1  1.500000e-89  XP_029241668.1   \n",
              "1                       1        1  4.800000e-88  XP_029239031.1   \n",
              "2                       2        2  9.900000e-04  XP_029242319.1   \n",
              "3                       1        1  2.100000e-45  XP_029233499.1   \n",
              "4                       1        1  4.000000e-44  XP_029240716.1   \n",
              "...                   ...      ...           ...             ...   \n",
              "3860                    1        1  1.200000e-03  XP_029242249.1   \n",
              "3861                    1        1  2.000000e-03  XP_029238453.1   \n",
              "3862                    1        1  5.300000e-02  XP_029234867.1   \n",
              "3863                    2        2  4.700000e+00  XP_029240503.1   \n",
              "3864                    2        2  8.600000e+00  XP_029240468.1   \n",
              "\n",
              "          query_id  overlap_num  region_num  \n",
              "0           14-3-3            0           1  \n",
              "1           14-3-3            0           1  \n",
              "2           14-3-3            0           2  \n",
              "3     2-Hacid_dh_C            0           1  \n",
              "4     2-Hacid_dh_C            0           1  \n",
              "...            ...          ...         ...  \n",
              "3860            ZZ            0           1  \n",
              "3861            ZZ            0           1  \n",
              "3862            ZZ            0           1  \n",
              "3863            ZZ            0           2  \n",
              "3864            ZZ            0           2  \n",
              "\n",
              "[3865 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4cd06c8c-4a0c-4686-8f46-3819618129d3\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>5.6</td>\n",
              "      <td>298.5</td>\n",
              "      <td>14-3-3 protein [Trypanosoma rangeli]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.500000e-89</td>\n",
              "      <td>XP_029241668.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>1.9</td>\n",
              "      <td>293.6</td>\n",
              "      <td>14-3-3 protein [Trypanosoma rangeli]</td>\n",
              "      <td>0</td>\n",
              "      <td>1.1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.800000e-88</td>\n",
              "      <td>XP_029239031.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>17.9</td>\n",
              "      <td>uncharacterized protein TraAM80_00771 [Trypano...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.5</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>9.900000e-04</td>\n",
              "      <td>XP_029242319.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>153.4</td>\n",
              "      <td>D-isomer specific 2-hydroxyacid dehydrogenase-...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.100000e-45</td>\n",
              "      <td>XP_029233499.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>149.3</td>\n",
              "      <td>D-isomer specific 2-hydroxyacid dehydrogenase-...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>4.000000e-44</td>\n",
              "      <td>XP_029240716.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3860</th>\n",
              "      <td>-</td>\n",
              "      <td>8.1</td>\n",
              "      <td>17.8</td>\n",
              "      <td>E3 ubiquitin-protein ligase TRIP12 [Trypanosom...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1.200000e-03</td>\n",
              "      <td>XP_029242249.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3861</th>\n",
              "      <td>-</td>\n",
              "      <td>11.8</td>\n",
              "      <td>17.1</td>\n",
              "      <td>uncharacterized protein TraAM80_04771 [Trypano...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.000000e-03</td>\n",
              "      <td>XP_029238453.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3862</th>\n",
              "      <td>-</td>\n",
              "      <td>1.5</td>\n",
              "      <td>12.5</td>\n",
              "      <td>ADP-ribosylation factor GTPase activating prot...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5.300000e-02</td>\n",
              "      <td>XP_029234867.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3863</th>\n",
              "      <td>-</td>\n",
              "      <td>11.1</td>\n",
              "      <td>6.3</td>\n",
              "      <td>putative protein kinase [Trypanosoma rangeli]</td>\n",
              "      <td>0</td>\n",
              "      <td>2.0</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>4.700000e+00</td>\n",
              "      <td>XP_029240503.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3864</th>\n",
              "      <td>-</td>\n",
              "      <td>13.1</td>\n",
              "      <td>5.5</td>\n",
              "      <td>other hect domain ubiquitin protein ligase E3 ...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.8</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>8.600000e+00</td>\n",
              "      <td>XP_029240468.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3865 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4cd06c8c-4a0c-4686-8f46-3819618129d3')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-4cd06c8c-4a0c-4686-8f46-3819618129d3 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-4cd06c8c-4a0c-4686-8f46-3819618129d3');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-58ae4d97-9f4a-4c79-984b-e168971e9e70\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-58ae4d97-9f4a-4c79-984b-e168971e9e70')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-58ae4d97-9f4a-4c79-984b-e168971e9e70 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 3865,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 9.316088816850714,\n        \"min\": 0.0,\n        \"max\": 175.0,\n        \"num_unique_values\": 324,\n        \"samples\": [\n          10.2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 66.89996481865306,\n        \"min\": 3.6,\n        \"max\": 876.5,\n        \"num_unique_values\": 1284,\n        \"samples\": [\n          87.8\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1554,\n        \"samples\": [\n          \"uncharacterized protein TraAM80_08840 [Trypanosoma rangeli]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 5,\n        \"num_unique_values\": 6,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.2844752886515,\n        \"min\": 1.0,\n        \"max\": 26.4,\n        \"num_unique_values\": 132,\n        \"samples\": [\n          4.7\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 18,\n        \"num_unique_values\": 18,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.7828995061701505,\n        \"min\": 4.1e-264,\n        \"max\": 9.8,\n        \"num_unique_values\": 1920,\n        \"samples\": [\n          0.00062\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2066,\n        \"samples\": [\n          \"XP_029238193.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 336,\n        \"samples\": [\n          \"DUSP\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 11,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          7\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 26,\n        \"num_unique_values\": 22,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 140
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Trypanosoma_rangeli_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "WRSrlMEQ73d0"
      },
      "execution_count": 141,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCF_003719475.1_ASM371947v1_genomic.fna -bed Trypanosoma_rangeli_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9a70a357-ef85-4f21-951d-3ab15a9309b3",
        "id": "0bzH8gHt73d0"
      },
      "execution_count": 142,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "index file GCF_003719475.1_ASM371947v1_genomic.fna.fai not found, generating...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_Trypanosoma_rangeli_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "rj2GYKS_73d1"
      },
      "execution_count": 143,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Trypanosoma brucei"
      ],
      "metadata": {
        "id": "w5co6wxAEuGh"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "maEC0-sBE3vz"
      },
      "execution_count": 144,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "hmmer_data = get_hmmer_data(\"Trypanosoma_brucei_filtered.out\")\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 808
        },
        "outputId": "b1dd47b5-45aa-4909-d81f-d32ee5efed56",
        "id": "A_35HJZqE3vz"
      },
      "execution_count": 145,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     accession  bias  bitscore  \\\n",
              "0            -   6.7     297.1   \n",
              "1            -   4.4     297.1   \n",
              "2            -   0.0      23.9   \n",
              "3            -   0.0      38.8   \n",
              "4            -   1.2      21.8   \n",
              "...        ...   ...       ...   \n",
              "3812         -   7.9      14.9   \n",
              "3813         -   6.5      13.3   \n",
              "3814         -   5.5      10.5   \n",
              "3815         -   6.3       8.5   \n",
              "3816         -  14.2       5.1   \n",
              "\n",
              "                                            description  cluster_num  \\\n",
              "0     14-3-3-like protein, putative [Trypanosoma bru...            0   \n",
              "1     14-3-3-like protein, putative [Trypanosoma bru...            0   \n",
              "2     hypothetical protein, conserved [Trypanosoma b...            0   \n",
              "3     S-adenosylhomocysteine hydrolase, putative [Tr...            0   \n",
              "4     2-hydroxy-3-oxopropionate reductase, putative ...            0   \n",
              "...                                                 ...          ...   \n",
              "3812  hypothetical protein, conserved [Trypanosoma b...            0   \n",
              "3813  protein kinase, putative [Trypanosoma brucei b...            0   \n",
              "3814  40S ribosomal protein S27, putative [Trypanoso...            1   \n",
              "3815  ubiquitin-protein ligase, putative [Trypanosom...            0   \n",
              "3816  ubiquitin-protein ligase, putative [Trypanosom...            0   \n",
              "\n",
              "      domain_exp_num  domain_included_num  domain_obs_num  \\\n",
              "0                1.0                    1               1   \n",
              "1                1.0                    1               1   \n",
              "2                2.2                    1               2   \n",
              "3                1.4                    1               1   \n",
              "4                2.7                    1               3   \n",
              "...              ...                  ...             ...   \n",
              "3812             2.7                    1               2   \n",
              "3813             1.9                    0               1   \n",
              "3814             2.4                    0               2   \n",
              "3815             1.6                    0               1   \n",
              "3816             2.2                    0               2   \n",
              "\n",
              "      domain_reported_num  env_num        evalue          id      query_id  \\\n",
              "0                       1        1  3.600000e-89  EAN79648.1        14-3-3   \n",
              "1                       1        1  3.600000e-89  EAN79905.1        14-3-3   \n",
              "2                       2        2  1.200000e-05  EAN78699.1        14-3-3   \n",
              "3                       1        1  2.800000e-10  EAN79911.1  2-Hacid_dh_C   \n",
              "4                       3        3  4.800000e-05  AAZ11720.1  2-Hacid_dh_C   \n",
              "...                   ...      ...           ...         ...           ...   \n",
              "3812                    2        2  8.600000e-03  EAN80571.1            ZZ   \n",
              "3813                    1        1  2.800000e-02  EAN80004.1            ZZ   \n",
              "3814                    2        2  2.000000e-01  EAN79924.1            ZZ   \n",
              "3815                    1        1  8.300000e-01  AAZ12927.1            ZZ   \n",
              "3816                    2        2  9.700000e+00  AAZ10668.1            ZZ   \n",
              "\n",
              "      overlap_num  region_num  \n",
              "0               0           1  \n",
              "1               0           1  \n",
              "2               0           2  \n",
              "3               0           1  \n",
              "4               0           3  \n",
              "...           ...         ...  \n",
              "3812            0           2  \n",
              "3813            0           1  \n",
              "3814            1           1  \n",
              "3815            0           1  \n",
              "3816            0           2  \n",
              "\n",
              "[3817 rows x 15 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-f977d8e4-3d27-40e9-99c0-05003c4a97c8\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>6.7</td>\n",
              "      <td>297.1</td>\n",
              "      <td>14-3-3-like protein, putative [Trypanosoma bru...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.600000e-89</td>\n",
              "      <td>EAN79648.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>4.4</td>\n",
              "      <td>297.1</td>\n",
              "      <td>14-3-3-like protein, putative [Trypanosoma bru...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3.600000e-89</td>\n",
              "      <td>EAN79905.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>23.9</td>\n",
              "      <td>hypothetical protein, conserved [Trypanosoma b...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>1.200000e-05</td>\n",
              "      <td>EAN78699.1</td>\n",
              "      <td>14-3-3</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>0.0</td>\n",
              "      <td>38.8</td>\n",
              "      <td>S-adenosylhomocysteine hydrolase, putative [Tr...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.4</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.800000e-10</td>\n",
              "      <td>EAN79911.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>1.2</td>\n",
              "      <td>21.8</td>\n",
              "      <td>2-hydroxy-3-oxopropionate reductase, putative ...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.7</td>\n",
              "      <td>1</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>4.800000e-05</td>\n",
              "      <td>AAZ11720.1</td>\n",
              "      <td>2-Hacid_dh_C</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3812</th>\n",
              "      <td>-</td>\n",
              "      <td>7.9</td>\n",
              "      <td>14.9</td>\n",
              "      <td>hypothetical protein, conserved [Trypanosoma b...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.7</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>8.600000e-03</td>\n",
              "      <td>EAN80571.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3813</th>\n",
              "      <td>-</td>\n",
              "      <td>6.5</td>\n",
              "      <td>13.3</td>\n",
              "      <td>protein kinase, putative [Trypanosoma brucei b...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.9</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>2.800000e-02</td>\n",
              "      <td>EAN80004.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3814</th>\n",
              "      <td>-</td>\n",
              "      <td>5.5</td>\n",
              "      <td>10.5</td>\n",
              "      <td>40S ribosomal protein S27, putative [Trypanoso...</td>\n",
              "      <td>1</td>\n",
              "      <td>2.4</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2.000000e-01</td>\n",
              "      <td>EAN79924.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3815</th>\n",
              "      <td>-</td>\n",
              "      <td>6.3</td>\n",
              "      <td>8.5</td>\n",
              "      <td>ubiquitin-protein ligase, putative [Trypanosom...</td>\n",
              "      <td>0</td>\n",
              "      <td>1.6</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>8.300000e-01</td>\n",
              "      <td>AAZ12927.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3816</th>\n",
              "      <td>-</td>\n",
              "      <td>14.2</td>\n",
              "      <td>5.1</td>\n",
              "      <td>ubiquitin-protein ligase, putative [Trypanosom...</td>\n",
              "      <td>0</td>\n",
              "      <td>2.2</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>9.700000e+00</td>\n",
              "      <td>AAZ10668.1</td>\n",
              "      <td>ZZ</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3817 rows × 15 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f977d8e4-3d27-40e9-99c0-05003c4a97c8')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-f977d8e4-3d27-40e9-99c0-05003c4a97c8 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-f977d8e4-3d27-40e9-99c0-05003c4a97c8');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-c8a022ee-c8e0-4602-92bd-e55d9c3963d5\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c8a022ee-c8e0-4602-92bd-e55d9c3963d5')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-c8a022ee-c8e0-4602-92bd-e55d9c3963d5 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 3817,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 12.773829129020099,\n        \"min\": 0.0,\n        \"max\": 170.4,\n        \"num_unique_values\": 362,\n        \"samples\": [\n          7.7\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 71.18345803424093,\n        \"min\": 3.3,\n        \"max\": 924.8,\n        \"num_unique_values\": 1232,\n        \"samples\": [\n          98.3\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 488,\n        \"samples\": [\n          \"RNA helicase, putative [Trypanosoma brucei brucei TREU927]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 9,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          9\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.8637904580808202,\n        \"min\": 1.0,\n        \"max\": 29.2,\n        \"num_unique_values\": 157,\n        \"samples\": [\n          22.4\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 19,\n        \"num_unique_values\": 19,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 36,\n        \"num_unique_values\": 33,\n        \"samples\": [\n          35\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 30,\n        \"num_unique_values\": 29,\n        \"samples\": [\n          12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 1,\n        \"max\": 36,\n        \"num_unique_values\": 33,\n        \"samples\": [\n          35\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.7815721745261214,\n        \"min\": 7.8e-287,\n        \"max\": 9.9,\n        \"num_unique_values\": 1865,\n        \"samples\": [\n          0.013\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1981,\n        \"samples\": [\n          \"EAN78807.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 326,\n        \"samples\": [\n          \"SF3b1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1,\n        \"min\": 0,\n        \"max\": 19,\n        \"num_unique_values\": 18,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 36,\n        \"num_unique_values\": 29,\n        \"samples\": [\n          26\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 145
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filename = \"Trypanosoma_brucei_hmr.bed\"\n",
        "f = open(filename, \"w\")\n",
        "\n",
        "for i in families:\n",
        "  gene = hmmer_data.loc[hmmer_data[\"query_id\"].str.contains(i)]\n",
        "  for index, row in gene.iterrows():\n",
        "    gene = genome.loc[genome[\"attributes\"].str.contains(row[\"id\"])]\n",
        "    for index, row in gene.iterrows():\n",
        "      f.write(f\"{row['chrom']}\\t{row['start']}\\t{row['end']}\\n\")\n",
        "      break\n",
        "    break\n",
        "\n",
        "f.close()"
      ],
      "metadata": {
        "id": "sfMjWlD3E3vz"
      },
      "execution_count": 146,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools getfasta -fi GCA_000002445.1_ASM244v1_genomic.fna -bed Trypanosoma_brucei_hmr.bed -fo genes_hmmer.fa"
      ],
      "metadata": {
        "id": "Bm2FPycWE3vz"
      },
      "execution_count": 148,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "o = open(\"genes_Trypanosoma_brucei_hmmer_cut.fa\", \"w\")\n",
        "with open(\"genes_hmmer.fa\", \"r\") as f:\n",
        "  while s := f.readline():\n",
        "    o.write(f\"{s.rstrip()[:2000]}\\n\")\n",
        "o.close()"
      ],
      "metadata": {
        "id": "ZiyKb9kzE3vz"
      },
      "execution_count": 149,
      "outputs": []
    }
  ]
}